{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f66a8ea9",
   "metadata": {},
   "source": [
    "# Create a VCF with additional alternative codons encoding the same amino acid. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f5071dc6",
   "metadata": {},
   "source": [
    "## 1. Preparation of the annotated file for step 3."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "48968a17",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import csv\n",
    "import allel\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c97ed3d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "vep_file = \"clinvar_vep_ref.vep\" # clinvar.vcf аннатированный VEP\n",
    "vcf_file = \"clinvar.vcf\" # https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/clinvar.vcf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a56cd734",
   "metadata": {},
   "source": [
    "Create a header for the clinvar_ann_df dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "aa63b1a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_header(file_name):\n",
    "    \"\"\"\n",
    "    saves the vep file header\n",
    "        :param file_name: vep file name\n",
    "    :return header_list[0]: header list\n",
    "    \"\"\"\n",
    "    \n",
    "    \n",
    "    header_list = []\n",
    "    with open(file_name, \"r\") as file:\n",
    "        while True:\n",
    "            line = file.readline()\n",
    "            if not line:\n",
    "                break\n",
    "            if line.split()[0] == \"#Uploaded_variation\": # сохраним заголовок\n",
    "                header_list.append(line.split())\n",
    "                return header_list[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "be34dc89",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['#Uploaded_variation',\n",
       " 'Location',\n",
       " 'Allele',\n",
       " 'Gene',\n",
       " 'Feature',\n",
       " 'Feature_type',\n",
       " 'Consequence',\n",
       " 'cDNA_position',\n",
       " 'CDS_position',\n",
       " 'Protein_position',\n",
       " 'Amino_acids',\n",
       " 'Codons',\n",
       " 'Existing_variation',\n",
       " 'Extra']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "header_list = make_header(vep_file)\n",
    "header_list"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3713678c",
   "metadata": {},
   "source": [
    "Download clinvar.vcf "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3be84492",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wall time: 17.6 s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>ALT_2</th>\n",
       "      <th>ALT_3</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>AF_ESP</th>\n",
       "      <th>AF_EXAC</th>\n",
       "      <th>...</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>RS</th>\n",
       "      <th>SSR</th>\n",
       "      <th>FILTER_PASS</th>\n",
       "      <th>numalt</th>\n",
       "      <th>altlen_1</th>\n",
       "      <th>altlen_2</th>\n",
       "      <th>altlen_3</th>\n",
       "      <th>is_snp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>925952</td>\n",
       "      <td>1019397</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>1640863258</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>925976</td>\n",
       "      <td>1362713</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>926003</td>\n",
       "      <td>1365270</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>926014</td>\n",
       "      <td>1377425</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>SO:0001575|splice_donor_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>930139</td>\n",
       "      <td>1125147</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>SO:0001627|intron_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57893</th>\n",
       "      <td>MT</td>\n",
       "      <td>16159</td>\n",
       "      <td>1525973</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57894</th>\n",
       "      <td>MT</td>\n",
       "      <td>16179</td>\n",
       "      <td>1525977</td>\n",
       "      <td>CAA</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>-2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57895</th>\n",
       "      <td>MT</td>\n",
       "      <td>16230</td>\n",
       "      <td>1525975</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57896</th>\n",
       "      <td>MT</td>\n",
       "      <td>16274</td>\n",
       "      <td>1525974</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57897</th>\n",
       "      <td>NW_009646201.1</td>\n",
       "      <td>83614</td>\n",
       "      <td>17735</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1556058284</td>\n",
       "      <td>-1</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1303082 rows × 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                CHROM     POS       ID  REF ALT_1  ALT_2  ALT_3  QUAL  AF_ESP  \\\n",
       "0                   1  925952  1019397    G     A    NaN    NaN   NaN     NaN   \n",
       "1                   1  925976  1362713    T     C    NaN    NaN   NaN     NaN   \n",
       "2                   1  926003  1365270    C     T    NaN    NaN   NaN     NaN   \n",
       "3                   1  926014  1377425    G     A    NaN    NaN   NaN     NaN   \n",
       "4                   1  930139  1125147    C     T    NaN    NaN   NaN     NaN   \n",
       "...               ...     ...      ...  ...   ...    ...    ...   ...     ...   \n",
       "57893              MT   16159  1525973    C     A    NaN    NaN   NaN     NaN   \n",
       "57894              MT   16179  1525977  CAA     C    NaN    NaN   NaN     NaN   \n",
       "57895              MT   16230  1525975    A     G    NaN    NaN   NaN     NaN   \n",
       "57896              MT   16274  1525974    G     A    NaN    NaN   NaN     NaN   \n",
       "57897  NW_009646201.1   83614    17735   TC     T    NaN    NaN   NaN     NaN   \n",
       "\n",
       "       AF_EXAC  ...                               MC  ORIGIN          RS SSR  \\\n",
       "0          NaN  ...      SO:0001583|missense_variant       1  1640863258  -1   \n",
       "1          NaN  ...      SO:0001583|missense_variant       1         NaN  -1   \n",
       "2          NaN  ...      SO:0001583|missense_variant       1         NaN  -1   \n",
       "3          NaN  ...  SO:0001575|splice_donor_variant       1         NaN  -1   \n",
       "4          NaN  ...        SO:0001627|intron_variant       1         NaN  -1   \n",
       "...        ...  ...                              ...     ...         ...  ..   \n",
       "57893      NaN  ...                              NaN      16         NaN  -1   \n",
       "57894      NaN  ...                              NaN      16         NaN  -1   \n",
       "57895      NaN  ...                              NaN      16         NaN  -1   \n",
       "57896      NaN  ...                              NaN      16         NaN  -1   \n",
       "57897      NaN  ...                              NaN       1  1556058284  -1   \n",
       "\n",
       "      FILTER_PASS numalt altlen_1 altlen_2 altlen_3 is_snp  \n",
       "0           False      1        0        0        0   True  \n",
       "1           False      1        0        0        0   True  \n",
       "2           False      1        0        0        0   True  \n",
       "3           False      1        0        0        0   True  \n",
       "4           False      1        0        0        0   True  \n",
       "...           ...    ...      ...      ...      ...    ...  \n",
       "57893       False      1        0        0        0   True  \n",
       "57894       False      1       -2        0        0  False  \n",
       "57895       False      1        0        0        0   True  \n",
       "57896       False      1        0        0        0   True  \n",
       "57897       False      1       -1        0        0  False  \n",
       "\n",
       "[1303082 rows x 36 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "df_clinvar = allel.vcf_to_dataframe(input = vcf_file, fields='*')\n",
    "df_clinvar"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4efa6dd3",
   "metadata": {},
   "source": [
    "Create clinvar_ann_df from clinvar_vep_ref.vep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7550b9d8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\Miniconda3\\lib\\site-packages\\IPython\\core\\magics\\execution.py:1324: DtypeWarning: Columns (3) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code, glob, local_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wall time: 58.3 s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>Extra</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1019397</td>\n",
       "      <td>1:925952</td>\n",
       "      <td>A</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385640.1</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1057</td>\n",
       "      <td>548</td>\n",
       "      <td>183</td>\n",
       "      <td>G/E</td>\n",
       "      <td>gGg/gAg</td>\n",
       "      <td>-</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;GIVEN_REF=G;USED_REF=G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1019397</td>\n",
       "      <td>1:925952</td>\n",
       "      <td>A</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385641.1</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1057</td>\n",
       "      <td>548</td>\n",
       "      <td>183</td>\n",
       "      <td>G/E</td>\n",
       "      <td>gGg/gAg</td>\n",
       "      <td>-</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1019397</td>\n",
       "      <td>1:925952</td>\n",
       "      <td>A</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_152486.4</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>101</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>G/E</td>\n",
       "      <td>gGg/gAg</td>\n",
       "      <td>-</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;GIVEN_REF=G;USED_REF=G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1019397</td>\n",
       "      <td>1:925952</td>\n",
       "      <td>A</td>\n",
       "      <td>107985728</td>\n",
       "      <td>NR_168405.1</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IMPACT=MODIFIER;DISTANCE=348;STRAND=-1;CANONIC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1362713</td>\n",
       "      <td>1:925976</td>\n",
       "      <td>C</td>\n",
       "      <td>148398</td>\n",
       "      <td>NM_001385640.1</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1081</td>\n",
       "      <td>572</td>\n",
       "      <td>191</td>\n",
       "      <td>I/T</td>\n",
       "      <td>aTc/aCc</td>\n",
       "      <td>rs1030245330</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;GIVEN_REF=T;USED_REF=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12858667</th>\n",
       "      <td>1525973</td>\n",
       "      <td>MT:16159</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IMPACT=MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12858668</th>\n",
       "      <td>1525977</td>\n",
       "      <td>MT:16180-16181</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs371240719</td>\n",
       "      <td>IMPACT=MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12858669</th>\n",
       "      <td>1525975</td>\n",
       "      <td>MT:16230</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs2853514</td>\n",
       "      <td>IMPACT=MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12858670</th>\n",
       "      <td>1525974</td>\n",
       "      <td>MT:16274</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs144095641</td>\n",
       "      <td>IMPACT=MODIFIER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12858671</th>\n",
       "      <td>17735</td>\n",
       "      <td>NW_009646201.1:83615</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>IMPACT=MODIFIER</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12858672 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          #Uploaded_variation              Location Allele       Gene  \\\n",
       "0                     1019397              1:925952      A     148398   \n",
       "1                     1019397              1:925952      A     148398   \n",
       "2                     1019397              1:925952      A     148398   \n",
       "3                     1019397              1:925952      A  107985728   \n",
       "4                     1362713              1:925976      C     148398   \n",
       "...                       ...                   ...    ...        ...   \n",
       "12858667              1525973              MT:16159      A          -   \n",
       "12858668              1525977        MT:16180-16181      -          -   \n",
       "12858669              1525975              MT:16230      G          -   \n",
       "12858670              1525974              MT:16274      A          -   \n",
       "12858671                17735  NW_009646201.1:83615      -          -   \n",
       "\n",
       "                 Feature Feature_type            Consequence cDNA_position  \\\n",
       "0         NM_001385640.1   Transcript       missense_variant          1057   \n",
       "1         NM_001385641.1   Transcript       missense_variant          1057   \n",
       "2            NM_152486.4   Transcript       missense_variant           101   \n",
       "3            NR_168405.1   Transcript  upstream_gene_variant             -   \n",
       "4         NM_001385640.1   Transcript       missense_variant          1081   \n",
       "...                  ...          ...                    ...           ...   \n",
       "12858667               -            -     intergenic_variant             -   \n",
       "12858668               -            -     intergenic_variant             -   \n",
       "12858669               -            -     intergenic_variant             -   \n",
       "12858670               -            -     intergenic_variant             -   \n",
       "12858671               -            -     intergenic_variant             -   \n",
       "\n",
       "         CDS_position Protein_position Amino_acids   Codons  \\\n",
       "0                 548              183         G/E  gGg/gAg   \n",
       "1                 548              183         G/E  gGg/gAg   \n",
       "2                  11                4         G/E  gGg/gAg   \n",
       "3                   -                -           -        -   \n",
       "4                 572              191         I/T  aTc/aCc   \n",
       "...               ...              ...         ...      ...   \n",
       "12858667            -                -           -        -   \n",
       "12858668            -                -           -        -   \n",
       "12858669            -                -           -        -   \n",
       "12858670            -                -           -        -   \n",
       "12858671            -                -           -        -   \n",
       "\n",
       "         Existing_variation                                              Extra  \n",
       "0                         -    IMPACT=MODERATE;STRAND=1;GIVEN_REF=G;USED_REF=G  \n",
       "1                         -  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "2                         -    IMPACT=MODERATE;STRAND=1;GIVEN_REF=G;USED_REF=G  \n",
       "3                         -  IMPACT=MODIFIER;DISTANCE=348;STRAND=-1;CANONIC...  \n",
       "4              rs1030245330  IMPACT=MODERATE;STRAND=1;GIVEN_REF=T;USED_REF=...  \n",
       "...                     ...                                                ...  \n",
       "12858667                  -                                    IMPACT=MODIFIER  \n",
       "12858668        rs371240719                                    IMPACT=MODIFIER  \n",
       "12858669          rs2853514                                    IMPACT=MODIFIER  \n",
       "12858670        rs144095641                                    IMPACT=MODIFIER  \n",
       "12858671                  -                                    IMPACT=MODIFIER  \n",
       "\n",
       "[12858672 rows x 14 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "clinvar_ann_df = pd.read_table(vep_file, header=None, comment='#')\n",
    "clinvar_ann_df.columns = header_list\n",
    "clinvar_ann_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9c3764ea",
   "metadata": {},
   "source": [
    "## 2. Extract OMIM genes from annotated clinvar (clinvar_vep_ref.vep)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "38312f5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "omim_list_file = 'omim_list.txt' # https://www.omim.org"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "88354d5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def omim_list_func(omim_list_file):\n",
    "    \"\"\"\n",
    "    create a list of genes\n",
    "        :param omim_list_file: file containing raw list of genes\n",
    "    :return omim_list_2: prepared list of genes\n",
    "    \"\"\"\n",
    "    \n",
    "    \n",
    "    omim_list = []\n",
    "    with open(omim_list_file, \"r\") as file: #создаем список генов\n",
    "        while True:\n",
    "            line = file.readline()\n",
    "            if not line:\n",
    "                break\n",
    "            omim_list.append(line)\n",
    "            omim_list_1 = [i.replace('\\n', '') for i in omim_list ]\n",
    "            omim_list_2 = [i for i in omim_list_1 if 'NA' not in i]\n",
    "    return omim_list_2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb1a2b6c",
   "metadata": {},
   "source": [
    "Removing NA and extra characters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9529ead7",
   "metadata": {},
   "outputs": [],
   "source": [
    "omim_list = omim_list_func(omim_list_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "59bff575",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Feature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NM_000014.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NM_144670.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NM_015665.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NM_024666.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NM_001605.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3945</th>\n",
       "      <td>NM_004773.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3946</th>\n",
       "      <td>NM_207341.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3947</th>\n",
       "      <td>NM_003460.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3948</th>\n",
       "      <td>NM_001110354.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3949</th>\n",
       "      <td>NM_020928.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3950 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Feature\n",
       "0        NM_000014.6\n",
       "1        NM_144670.6\n",
       "2        NM_015665.6\n",
       "3        NM_024666.5\n",
       "4        NM_001605.3\n",
       "...              ...\n",
       "3945     NM_004773.4\n",
       "3946     NM_207341.3\n",
       "3947     NM_003460.2\n",
       "3948  NM_001110354.2\n",
       "3949     NM_020928.2\n",
       "\n",
       "[3950 rows x 1 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omim_pd = pd.DataFrame(omim_list, columns=['Feature'])\n",
    "omim_pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "226b8295",
   "metadata": {},
   "source": [
    "Extract strings from clinvar_vep_ref.vep by OMIM features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9313e80d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>Extra</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>713107.0</td>\n",
       "      <td>12:9074556</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>splice_donor_region_variant,intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs191120289</td>\n",
       "      <td>IMPACT=LOW;STRAND=-1;CANONICAL=YES;GIVEN_REF=C...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>781242.0</td>\n",
       "      <td>12:9076832</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>3526</td>\n",
       "      <td>3456</td>\n",
       "      <td>1152</td>\n",
       "      <td>Y</td>\n",
       "      <td>taT/taC</td>\n",
       "      <td>rs372050343,COSV59387215</td>\n",
       "      <td>IMPACT=LOW;STRAND=-1;CANONICAL=YES;GIVEN_REF=A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>711716.0</td>\n",
       "      <td>12:9079271</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3162</td>\n",
       "      <td>3092</td>\n",
       "      <td>1031</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGa/cAa</td>\n",
       "      <td>rs202210880,COSV59389075</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>726982.0</td>\n",
       "      <td>12:9079279</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>3154</td>\n",
       "      <td>3084</td>\n",
       "      <td>1028</td>\n",
       "      <td>F</td>\n",
       "      <td>ttT/ttC</td>\n",
       "      <td>rs769231051</td>\n",
       "      <td>IMPACT=LOW;STRAND=-1;CANONICAL=YES;GIVEN_REF=A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>18171.0</td>\n",
       "      <td>12:9079672</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3068</td>\n",
       "      <td>2998</td>\n",
       "      <td>1000</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>rs669,CM980001,COSV59386644</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1503659</th>\n",
       "      <td>155772.0</td>\n",
       "      <td>5:61544156</td>\n",
       "      <td>T</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3502</td>\n",
       "      <td>3487</td>\n",
       "      <td>1163</td>\n",
       "      <td>R/W</td>\n",
       "      <td>Cgg/Tgg</td>\n",
       "      <td>rs587777695,CM147511</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1503660</th>\n",
       "      <td>450504.0</td>\n",
       "      <td>5:61544160</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3506</td>\n",
       "      <td>3491</td>\n",
       "      <td>1164</td>\n",
       "      <td>H/R</td>\n",
       "      <td>cAc/cGc</td>\n",
       "      <td>rs1554041457</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1503661</th>\n",
       "      <td>1424334.0</td>\n",
       "      <td>5:61544165</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3511</td>\n",
       "      <td>3496</td>\n",
       "      <td>1166</td>\n",
       "      <td>S/G</td>\n",
       "      <td>Agt/Ggt</td>\n",
       "      <td>-</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1503662</th>\n",
       "      <td>1417067.0</td>\n",
       "      <td>5:61544201</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3547</td>\n",
       "      <td>3532</td>\n",
       "      <td>1178</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Acc/Gcc</td>\n",
       "      <td>rs1203405019</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1503663</th>\n",
       "      <td>1353194.0</td>\n",
       "      <td>5:61544238</td>\n",
       "      <td>T</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3584</td>\n",
       "      <td>3569</td>\n",
       "      <td>1190</td>\n",
       "      <td>T/I</td>\n",
       "      <td>aCa/aTa</td>\n",
       "      <td>rs1436393669</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1503664 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #Uploaded_variation    Location Allele   Gene      Feature  \\\n",
       "0                   713107.0  12:9074556      T      2  NM_000014.6   \n",
       "1                   781242.0  12:9076832      G      2  NM_000014.6   \n",
       "2                   711716.0  12:9079271      T      2  NM_000014.6   \n",
       "3                   726982.0  12:9079279      G      2  NM_000014.6   \n",
       "4                    18171.0  12:9079672      C      2  NM_000014.6   \n",
       "...                      ...         ...    ...    ...          ...   \n",
       "1503659             155772.0  5:61544156      T  57688  NM_020928.2   \n",
       "1503660             450504.0  5:61544160      G  57688  NM_020928.2   \n",
       "1503661            1424334.0  5:61544165      G  57688  NM_020928.2   \n",
       "1503662            1417067.0  5:61544201      G  57688  NM_020928.2   \n",
       "1503663            1353194.0  5:61544238      T  57688  NM_020928.2   \n",
       "\n",
       "        Feature_type                                 Consequence  \\\n",
       "0         Transcript  splice_donor_region_variant,intron_variant   \n",
       "1         Transcript                          synonymous_variant   \n",
       "2         Transcript                            missense_variant   \n",
       "3         Transcript                          synonymous_variant   \n",
       "4         Transcript                            missense_variant   \n",
       "...              ...                                         ...   \n",
       "1503659   Transcript                            missense_variant   \n",
       "1503660   Transcript                            missense_variant   \n",
       "1503661   Transcript                            missense_variant   \n",
       "1503662   Transcript                            missense_variant   \n",
       "1503663   Transcript                            missense_variant   \n",
       "\n",
       "        cDNA_position CDS_position Protein_position Amino_acids   Codons  \\\n",
       "0                   -            -                -           -        -   \n",
       "1                3526         3456             1152           Y  taT/taC   \n",
       "2                3162         3092             1031         R/Q  cGa/cAa   \n",
       "3                3154         3084             1028           F  ttT/ttC   \n",
       "4                3068         2998             1000         I/V  Atc/Gtc   \n",
       "...               ...          ...              ...         ...      ...   \n",
       "1503659          3502         3487             1163         R/W  Cgg/Tgg   \n",
       "1503660          3506         3491             1164         H/R  cAc/cGc   \n",
       "1503661          3511         3496             1166         S/G  Agt/Ggt   \n",
       "1503662          3547         3532             1178         T/A  Acc/Gcc   \n",
       "1503663          3584         3569             1190         T/I  aCa/aTa   \n",
       "\n",
       "                  Existing_variation  \\\n",
       "0                        rs191120289   \n",
       "1           rs372050343,COSV59387215   \n",
       "2           rs202210880,COSV59389075   \n",
       "3                        rs769231051   \n",
       "4        rs669,CM980001,COSV59386644   \n",
       "...                              ...   \n",
       "1503659         rs587777695,CM147511   \n",
       "1503660                 rs1554041457   \n",
       "1503661                            -   \n",
       "1503662                 rs1203405019   \n",
       "1503663                 rs1436393669   \n",
       "\n",
       "                                                     Extra  \n",
       "0        IMPACT=LOW;STRAND=-1;CANONICAL=YES;GIVEN_REF=C...  \n",
       "1        IMPACT=LOW;STRAND=-1;CANONICAL=YES;GIVEN_REF=A...  \n",
       "2        IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "3        IMPACT=LOW;STRAND=-1;CANONICAL=YES;GIVEN_REF=A...  \n",
       "4        IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "...                                                    ...  \n",
       "1503659  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "1503660  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "1503661  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "1503662  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "1503663  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "\n",
       "[1503664 rows x 14 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_omim_df = clinvar_ann_df.merge(omim_pd, on='Feature', how='right')\n",
    "search_omim_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d88fdb3",
   "metadata": {},
   "source": [
    "Select canonical and missense variants only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "bcfbe33c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\1\\AppData\\Local\\Temp/ipykernel_19652/3844229404.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  omim_missence_canonical_df['#Uploaded_variation']=omim_missence_canonical_df['#Uploaded_variation'].astype(int)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>Extra</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>711716</td>\n",
       "      <td>12:9079271</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3162</td>\n",
       "      <td>3092</td>\n",
       "      <td>1031</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGa/cAa</td>\n",
       "      <td>rs202210880,COSV59389075</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>18171</td>\n",
       "      <td>12:9079672</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3068</td>\n",
       "      <td>2998</td>\n",
       "      <td>1000</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>rs669,CM980001,COSV59386644</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>18172</td>\n",
       "      <td>12:9079755</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2985</td>\n",
       "      <td>2915</td>\n",
       "      <td>972</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>tGt/tAt</td>\n",
       "      <td>rs1800433,CM920001</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>18174</td>\n",
       "      <td>12:9094987</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2181</td>\n",
       "      <td>2111</td>\n",
       "      <td>704</td>\n",
       "      <td>R/H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>rs1800434,COSV59390885,COSV59390935</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>729811</td>\n",
       "      <td>12:9107574</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>899</td>\n",
       "      <td>829</td>\n",
       "      <td>277</td>\n",
       "      <td>D/N</td>\n",
       "      <td>Gac/Aac</td>\n",
       "      <td>rs140258425</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1503659</th>\n",
       "      <td>155772</td>\n",
       "      <td>5:61544156</td>\n",
       "      <td>T</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3502</td>\n",
       "      <td>3487</td>\n",
       "      <td>1163</td>\n",
       "      <td>R/W</td>\n",
       "      <td>Cgg/Tgg</td>\n",
       "      <td>rs587777695,CM147511</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1503660</th>\n",
       "      <td>450504</td>\n",
       "      <td>5:61544160</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3506</td>\n",
       "      <td>3491</td>\n",
       "      <td>1164</td>\n",
       "      <td>H/R</td>\n",
       "      <td>cAc/cGc</td>\n",
       "      <td>rs1554041457</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1503661</th>\n",
       "      <td>1424334</td>\n",
       "      <td>5:61544165</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3511</td>\n",
       "      <td>3496</td>\n",
       "      <td>1166</td>\n",
       "      <td>S/G</td>\n",
       "      <td>Agt/Ggt</td>\n",
       "      <td>-</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1503662</th>\n",
       "      <td>1417067</td>\n",
       "      <td>5:61544201</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3547</td>\n",
       "      <td>3532</td>\n",
       "      <td>1178</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Acc/Gcc</td>\n",
       "      <td>rs1203405019</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1503663</th>\n",
       "      <td>1353194</td>\n",
       "      <td>5:61544238</td>\n",
       "      <td>T</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3584</td>\n",
       "      <td>3569</td>\n",
       "      <td>1190</td>\n",
       "      <td>T/I</td>\n",
       "      <td>aCa/aTa</td>\n",
       "      <td>rs1436393669</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>432380 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #Uploaded_variation    Location Allele   Gene      Feature  \\\n",
       "2                     711716  12:9079271      T      2  NM_000014.6   \n",
       "4                      18171  12:9079672      C      2  NM_000014.6   \n",
       "5                      18172  12:9079755      T      2  NM_000014.6   \n",
       "11                     18174  12:9094987      T      2  NM_000014.6   \n",
       "15                    729811  12:9107574      T      2  NM_000014.6   \n",
       "...                      ...         ...    ...    ...          ...   \n",
       "1503659               155772  5:61544156      T  57688  NM_020928.2   \n",
       "1503660               450504  5:61544160      G  57688  NM_020928.2   \n",
       "1503661              1424334  5:61544165      G  57688  NM_020928.2   \n",
       "1503662              1417067  5:61544201      G  57688  NM_020928.2   \n",
       "1503663              1353194  5:61544238      T  57688  NM_020928.2   \n",
       "\n",
       "        Feature_type       Consequence cDNA_position CDS_position  \\\n",
       "2         Transcript  missense_variant          3162         3092   \n",
       "4         Transcript  missense_variant          3068         2998   \n",
       "5         Transcript  missense_variant          2985         2915   \n",
       "11        Transcript  missense_variant          2181         2111   \n",
       "15        Transcript  missense_variant           899          829   \n",
       "...              ...               ...           ...          ...   \n",
       "1503659   Transcript  missense_variant          3502         3487   \n",
       "1503660   Transcript  missense_variant          3506         3491   \n",
       "1503661   Transcript  missense_variant          3511         3496   \n",
       "1503662   Transcript  missense_variant          3547         3532   \n",
       "1503663   Transcript  missense_variant          3584         3569   \n",
       "\n",
       "        Protein_position Amino_acids   Codons  \\\n",
       "2                   1031         R/Q  cGa/cAa   \n",
       "4                   1000         I/V  Atc/Gtc   \n",
       "5                    972         C/Y  tGt/tAt   \n",
       "11                   704         R/H  cGt/cAt   \n",
       "15                   277         D/N  Gac/Aac   \n",
       "...                  ...         ...      ...   \n",
       "1503659             1163         R/W  Cgg/Tgg   \n",
       "1503660             1164         H/R  cAc/cGc   \n",
       "1503661             1166         S/G  Agt/Ggt   \n",
       "1503662             1178         T/A  Acc/Gcc   \n",
       "1503663             1190         T/I  aCa/aTa   \n",
       "\n",
       "                          Existing_variation  \\\n",
       "2                   rs202210880,COSV59389075   \n",
       "4                rs669,CM980001,COSV59386644   \n",
       "5                         rs1800433,CM920001   \n",
       "11       rs1800434,COSV59390885,COSV59390935   \n",
       "15                               rs140258425   \n",
       "...                                      ...   \n",
       "1503659                 rs587777695,CM147511   \n",
       "1503660                         rs1554041457   \n",
       "1503661                                    -   \n",
       "1503662                         rs1203405019   \n",
       "1503663                         rs1436393669   \n",
       "\n",
       "                                                     Extra  \n",
       "2        IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "4        IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "5        IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "11       IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "15       IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "...                                                    ...  \n",
       "1503659  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "1503660  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "1503661  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "1503662  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "1503663  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "\n",
       "[432380 rows x 14 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "omim_missence_canonical_df = search_omim_df[(search_omim_df['Consequence'] == 'missense_variant') & (search_omim_df['Extra'].str.contains(\"CANONICAL=YES\")==True)]\n",
    "omim_missence_canonical_df['#Uploaded_variation']=omim_missence_canonical_df['#Uploaded_variation'].astype(int)\n",
    "omim_missence_canonical_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f6d7b393",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>Extra</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>711716</td>\n",
       "      <td>12:9079271</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3162</td>\n",
       "      <td>3092</td>\n",
       "      <td>1031</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGa/cAa</td>\n",
       "      <td>rs202210880,COSV59389075</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18171</td>\n",
       "      <td>12:9079672</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3068</td>\n",
       "      <td>2998</td>\n",
       "      <td>1000</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>rs669,CM980001,COSV59386644</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>18172</td>\n",
       "      <td>12:9079755</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2985</td>\n",
       "      <td>2915</td>\n",
       "      <td>972</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>tGt/tAt</td>\n",
       "      <td>rs1800433,CM920001</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>18174</td>\n",
       "      <td>12:9094987</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2181</td>\n",
       "      <td>2111</td>\n",
       "      <td>704</td>\n",
       "      <td>R/H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>rs1800434,COSV59390885,COSV59390935</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>729811</td>\n",
       "      <td>12:9107574</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>899</td>\n",
       "      <td>829</td>\n",
       "      <td>277</td>\n",
       "      <td>D/N</td>\n",
       "      <td>Gac/Aac</td>\n",
       "      <td>rs140258425</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432375</th>\n",
       "      <td>155772</td>\n",
       "      <td>5:61544156</td>\n",
       "      <td>T</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3502</td>\n",
       "      <td>3487</td>\n",
       "      <td>1163</td>\n",
       "      <td>R/W</td>\n",
       "      <td>Cgg/Tgg</td>\n",
       "      <td>rs587777695,CM147511</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432376</th>\n",
       "      <td>450504</td>\n",
       "      <td>5:61544160</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3506</td>\n",
       "      <td>3491</td>\n",
       "      <td>1164</td>\n",
       "      <td>H/R</td>\n",
       "      <td>cAc/cGc</td>\n",
       "      <td>rs1554041457</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432377</th>\n",
       "      <td>1424334</td>\n",
       "      <td>5:61544165</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3511</td>\n",
       "      <td>3496</td>\n",
       "      <td>1166</td>\n",
       "      <td>S/G</td>\n",
       "      <td>Agt/Ggt</td>\n",
       "      <td>-</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432378</th>\n",
       "      <td>1417067</td>\n",
       "      <td>5:61544201</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3547</td>\n",
       "      <td>3532</td>\n",
       "      <td>1178</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Acc/Gcc</td>\n",
       "      <td>rs1203405019</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432379</th>\n",
       "      <td>1353194</td>\n",
       "      <td>5:61544238</td>\n",
       "      <td>T</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3584</td>\n",
       "      <td>3569</td>\n",
       "      <td>1190</td>\n",
       "      <td>T/I</td>\n",
       "      <td>aCa/aTa</td>\n",
       "      <td>rs1436393669</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>432380 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #Uploaded_variation    Location Allele   Gene      Feature  \\\n",
       "0                    711716  12:9079271      T      2  NM_000014.6   \n",
       "1                     18171  12:9079672      C      2  NM_000014.6   \n",
       "2                     18172  12:9079755      T      2  NM_000014.6   \n",
       "3                     18174  12:9094987      T      2  NM_000014.6   \n",
       "4                    729811  12:9107574      T      2  NM_000014.6   \n",
       "...                     ...         ...    ...    ...          ...   \n",
       "432375               155772  5:61544156      T  57688  NM_020928.2   \n",
       "432376               450504  5:61544160      G  57688  NM_020928.2   \n",
       "432377              1424334  5:61544165      G  57688  NM_020928.2   \n",
       "432378              1417067  5:61544201      G  57688  NM_020928.2   \n",
       "432379              1353194  5:61544238      T  57688  NM_020928.2   \n",
       "\n",
       "       Feature_type       Consequence cDNA_position CDS_position  \\\n",
       "0        Transcript  missense_variant          3162         3092   \n",
       "1        Transcript  missense_variant          3068         2998   \n",
       "2        Transcript  missense_variant          2985         2915   \n",
       "3        Transcript  missense_variant          2181         2111   \n",
       "4        Transcript  missense_variant           899          829   \n",
       "...             ...               ...           ...          ...   \n",
       "432375   Transcript  missense_variant          3502         3487   \n",
       "432376   Transcript  missense_variant          3506         3491   \n",
       "432377   Transcript  missense_variant          3511         3496   \n",
       "432378   Transcript  missense_variant          3547         3532   \n",
       "432379   Transcript  missense_variant          3584         3569   \n",
       "\n",
       "       Protein_position Amino_acids   Codons  \\\n",
       "0                  1031         R/Q  cGa/cAa   \n",
       "1                  1000         I/V  Atc/Gtc   \n",
       "2                   972         C/Y  tGt/tAt   \n",
       "3                   704         R/H  cGt/cAt   \n",
       "4                   277         D/N  Gac/Aac   \n",
       "...                 ...         ...      ...   \n",
       "432375             1163         R/W  Cgg/Tgg   \n",
       "432376             1164         H/R  cAc/cGc   \n",
       "432377             1166         S/G  Agt/Ggt   \n",
       "432378             1178         T/A  Acc/Gcc   \n",
       "432379             1190         T/I  aCa/aTa   \n",
       "\n",
       "                         Existing_variation  \\\n",
       "0                  rs202210880,COSV59389075   \n",
       "1               rs669,CM980001,COSV59386644   \n",
       "2                        rs1800433,CM920001   \n",
       "3       rs1800434,COSV59390885,COSV59390935   \n",
       "4                               rs140258425   \n",
       "...                                     ...   \n",
       "432375                 rs587777695,CM147511   \n",
       "432376                         rs1554041457   \n",
       "432377                                    -   \n",
       "432378                         rs1203405019   \n",
       "432379                         rs1436393669   \n",
       "\n",
       "                                                    Extra  \n",
       "0       IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "1       IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "2       IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "3       IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "4       IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "...                                                   ...  \n",
       "432375  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "432376  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "432377  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "432378  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "432379  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "\n",
       "[432380 rows x 14 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_1 = omim_missence_canonical_df.reset_index(drop=True)\n",
    "df_1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8863adf",
   "metadata": {},
   "source": [
    "Take the columns: #Uploaded_variation, Allele, Gene, cDNA_position, CDS_position, Protein_position, Amino_acids, Codons from df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "53da2888",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_2 = df_1.iloc[:, [0,2,3,7,8,9,10,11]]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "80508dee",
   "metadata": {},
   "source": [
    "Create lists ref_amino_list, alt_amino_list, ref_codon_list, alt_codon_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "65ce18be",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_lists(df):\n",
    "    \"\"\"\n",
    "    an iterator that creates the necessary lists\n",
    "        :param df: dataframe\n",
    "    :yield ref_amino: reference amino acid\n",
    "    :yield alt_amino: atlernative amino acid\n",
    "    :yield ref_codon: reference codon\n",
    "    :yield alt_codon: atlernative codon\n",
    "    \"\"\"\n",
    "    \n",
    "    \n",
    "    for i, row in df.iterrows():\n",
    "        ref_amino = row[6][:1]\n",
    "        alt_amino = row[6][2:]\n",
    "        ref_codon = row[7][:3]\n",
    "        alt_codon = row[7][4:]\n",
    "        yield ref_amino, alt_amino, ref_codon, alt_codon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "acfb3027",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wall time: 18.8 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "generator = make_lists(df_2)\n",
    "all_list = []\n",
    "ref_amino_list = []\n",
    "alt_amino_list = []\n",
    "ref_codon_list = []\n",
    "alt_codon_list = []\n",
    "i = 0\n",
    "while True:\n",
    "    try:\n",
    "        all_list.append(next(generator))\n",
    "        ref_amino_list.append(all_list[i][0])\n",
    "        alt_amino_list.append(all_list[i][1])\n",
    "        ref_codon_list.append(all_list[i][2])\n",
    "        alt_codon_list.append(all_list[i][3])\n",
    "        i+=1\n",
    "    except StopIteration:\n",
    "        break"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9166de73",
   "metadata": {},
   "source": [
    "## 3. Creating an alternative codon that codes for the same amino acid, differs by one nucleotide and was not previously presented"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09f8897b",
   "metadata": {},
   "source": [
    "Create dictionaries of codons and amino acids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "b89ab763",
   "metadata": {},
   "outputs": [],
   "source": [
    "codon_table = {\n",
    "    'A': ('GCT', 'GCC', 'GCA', 'GCG'),\n",
    "    'C': ('TGT', 'TGC'),\n",
    "    'D': ('GAT', 'GAC'),\n",
    "    'E': ('GAA', 'GAG'),\n",
    "    'F': ('TTT', 'TTC'),\n",
    "    'G': ('GGT', 'GGC', 'GGA', 'GGG'),\n",
    "    'I': ('ATT', 'ATC', 'ATA'),\n",
    "    'H': ('CAT', 'CAC'),\n",
    "    'K': ('AAA', 'AAG'),\n",
    "    'L': ('TTA', 'TTG', 'CTT', 'CTC', 'CTA', 'CTG'),\n",
    "    'M': ('ATG',),\n",
    "    'N': ('AAT', 'AAC'),\n",
    "    'P': ('CCT', 'CCC', 'CCA', 'CCG'),\n",
    "    'Q': ('CAA', 'CAG'),\n",
    "    'R': ('CGT', 'CGC', 'CGA', 'CGG', 'AGA', 'AGG'),\n",
    "    'S': ('TCT', 'TCC', 'TCA', 'TCG', 'AGT', 'AGC'),\n",
    "    'T': ('ACT', 'ACC', 'ACA', 'ACG'),\n",
    "    'V': ('GTT', 'GTC', 'GTA', 'GTG'),\n",
    "    'W': ('TGG',),\n",
    "    'Y': ('TAT', 'TAC'),\n",
    "    '*': ('TAA', 'TAG', 'TGA'),\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bbcdb272",
   "metadata": {},
   "source": [
    "Create the *compare_triple* codon comparison function. If the codons differ by one nucleatide, outputs the reference codon, alternative codon, reference nucleotide, alternative nucleotide, and the substitution position in the codon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "1866b71d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def compare_triple(triple_a, triple_b):\n",
    "    \"\"\"\n",
    "    сompare two triples, if they differ by one letter\n",
    "        :param triple_a: string of first triples\n",
    "        :param triple_b: string of second triples\n",
    "    :return tuple of triple_a: string of first triples\n",
    "        :triple_b: string of second triples\n",
    "        :list_dif_nucl[0][0]: reference nucleotide\n",
    "        :list_dif_nucl[0][1]: atlernative nucleotide \n",
    "        :list_dif_nucl[0][2]: position\n",
    "    \"\"\"\n",
    "\n",
    "\n",
    "    list_dif_nucl = []\n",
    "    pos = 0\n",
    "    for x, y in zip(triple_a, triple_b):\n",
    "        pos += 1\n",
    "        if x != y:\n",
    "            list_dif_nucl.append([x, y, pos])\n",
    "    if len(list_dif_nucl) == 1:\n",
    "        return (\n",
    "            triple_a,\n",
    "            triple_b,\n",
    "            list_dif_nucl[0][0],\n",
    "            list_dif_nucl[0][1],\n",
    "            list_dif_nucl[0][2],\n",
    "        )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "670c9a1d",
   "metadata": {},
   "source": [
    "Create a function to compare a reference codon with a tuple of alternative codons *compare_ref_with_codons*. It sends the reference codon and the alternate codon from the *compare_lists* function to the *compare_triple* function. *compare_ref_with_codons* excludes an alternative codon that is already present in the clinvar_vep_ref.vcf file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e28e5045",
   "metadata": {},
   "outputs": [],
   "source": [
    "def compare_ref_with_codons(ref_codon, alt_codons, alt_codon):\n",
    "    \"\"\"\n",
    "    compares the reference codon with alternative codons\n",
    "        :param ref_codon: reference codon string\n",
    "        :param alt_codons: list of alternative codons\n",
    "        :param alt_codon: alternative codon string \n",
    "    :return list_compare_triple: a list containing the string of first triples\n",
    "        string of second triples\n",
    "        nucleotide reference\n",
    "        atlernative nucleotide\n",
    "        position\n",
    "    \"\"\"\n",
    "\n",
    "    \n",
    "    list_compare_triple = []\n",
    "    ref_pos = 1\n",
    "    alt_pos = 1\n",
    "    for i in alt_codons:\n",
    "        list_a = []\n",
    "        dif = 0\n",
    "        if compare_triple(ref_codon, i) and i == alt_codon:\n",
    "            ref_pos = compare_triple(ref_codon, i)[4]\n",
    "    for i in alt_codons:\n",
    "        list_a = []\n",
    "        if compare_triple(ref_codon, i) and i != alt_codon:\n",
    "            alt_pos = compare_triple(ref_codon, i)[4]\n",
    "            list_a.append(compare_triple(ref_codon, i)[0:4])\n",
    "            dif = alt_pos - ref_pos\n",
    "            list_a.append(dif)    \n",
    "            list_compare_triple.append(list_a)\n",
    "    return list_compare_triple"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38a05af5",
   "metadata": {},
   "source": [
    "Create a function that generates a list of additional alternative codons encoding the same amino acid that are not in the source file. It takes as input lists of reference codons, alternative amino acids, alternative codons, which are taken from *clinvar_vep_ref.vcf*."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "28aecac1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def compare_lists(ref_codon_list, alt_amino_list, alt_codon_list):\n",
    "    \"\"\"\n",
    "    compares the list of reference codons with the list of alternative codons\n",
    "        :param ref_codon_list: reference codons list\n",
    "        :param alt_amino_list: list of alternative amino acids\n",
    "        :param alt_codon_list: alternative codons list \n",
    "    :return additional_amino_list: a list containing the lists needed for the dataframe\n",
    "    \"\"\"\n",
    "\n",
    "    \n",
    "    additional_amino_list = []\n",
    "    for ref_codon, alt_amino, alt_codon in zip(ref_codon_list, alt_amino_list, alt_codon_list):\n",
    "        try:\n",
    "            additional_amino_list.append(compare_ref_with_codons(ref_codon.upper(), codon_table[alt_amino], alt_codon.upper())) # сравниваем референсный кодон с кортежем альтернативных кодонов  \n",
    "        except KeyError:\n",
    "            additional_amino_list.append('#') # If a key error is marked with \"#\"\n",
    "            pass\n",
    "    return additional_amino_list"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d37efc76",
   "metadata": {},
   "source": [
    "Run the *compare_lists function* and create an *additional_amino_list*."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "fdfbb16a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAG', 'GAC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GTG', 'CTG', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('CAG', 'CAC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'ATT', 'G', 'T'), 0], [('ATG', 'ATA', 'G', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AGT', 'CGT', 'A', 'C'), -2], [('AGT', 'AGA', 'T', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'TTG', 'A', 'T'), 0]],\n",
       " [],\n",
       " [[('GTG', 'TTG', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [[('TGG', 'AGG', 'T', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAC', 'GAG', 'C', 'G'), 0]],\n",
       " [],\n",
       " [[('GGA', 'CGA', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGA', 'CGA', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGG', 'CGG', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AGC', 'CGC', 'A', 'C'), -2], [('AGC', 'AGA', 'C', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TTT', 'TTA', 'T', 'A'), 2], [('TTT', 'TTG', 'T', 'G'), 2]],\n",
       " [],\n",
       " [],\n",
       " [[('TGG', 'TGT', 'G', 'T'), 0]],\n",
       " [[('AAT', 'AAG', 'T', 'G'), 0]],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'CTG', 'A', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('CAA', 'CAT', 'A', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAG', 'GAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'ATT', 'G', 'T'), 0], [('ATG', 'ATC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TGG', 'AGG', 'T', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAC', 'GAG', 'C', 'G'), 0]],\n",
       " [],\n",
       " [[('TTC', 'TTA', 'C', 'A'), 2], [('TTC', 'TTG', 'C', 'G'), 2]],\n",
       " [],\n",
       " [[('GTG', 'CTG', 'G', 'C'), 0]],\n",
       " [],\n",
       " '#',\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TTC', 'TTA', 'C', 'A'), 2], [('TTC', 'TTG', 'C', 'G'), 2]],\n",
       " [],\n",
       " [],\n",
       " [[('AAC', 'AAA', 'C', 'A'), 0]],\n",
       " [[('GGG', 'CGG', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAC', 'GAG', 'C', 'G'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGG', 'CGG', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [[('GAC', 'GAA', 'C', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TTT', 'TTA', 'T', 'A'), 2], [('TTT', 'TTG', 'T', 'G'), 2]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TGG', 'AGG', 'T', 'A'), 0]],\n",
       " [[('TGG', 'TGT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAC', 'GAG', 'C', 'G'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AGT', 'AGA', 'T', 'A'), 2], [('AGT', 'AGG', 'T', 'G'), 2]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('CAA', 'CAC', 'A', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGG', 'AGG', 'G', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GTG', 'TTG', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'TTG', 'A', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TTC', 'TTA', 'C', 'A'), 2], [('TTC', 'TTG', 'C', 'G'), 2]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TGG', 'AGG', 'T', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGG', 'CGG', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AAC', 'AAA', 'C', 'A'), 0]],\n",
       " [[('CAG', 'CAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('CAG', 'CAC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGA', 'CGA', 'G', 'C'), 0]],\n",
       " [],\n",
       " [[('ATG', 'ATT', 'G', 'T'), 0], [('ATG', 'ATC', 'G', 'C'), 0]],\n",
       " [[('GAG', 'GAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGA', 'AGA', 'G', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AGC', 'CGC', 'A', 'C'), -2], [('AGC', 'AGA', 'C', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GTG', 'CTG', 'G', 'C'), 0]],\n",
       " [[('ATG', 'ATT', 'G', 'T'), 0], [('ATG', 'ATC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGG', 'CGG', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAA', 'GAT', 'A', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TGG', 'TGT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GTG', 'TTG', 'G', 'T'), 0]],\n",
       " [],\n",
       " [[('TGC', 'AGC', 'T', 'A'), -1]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TTC', 'TTG', 'C', 'G'), 0], [('TTC', 'CTC', 'T', 'C'), -2]],\n",
       " [],\n",
       " [[('TTT', 'TTA', 'T', 'A'), 0], [('TTT', 'CTT', 'T', 'C'), -2]],\n",
       " [[('GTG', 'CTG', 'G', 'C'), 0]],\n",
       " [[('GAT', 'GAG', 'T', 'G'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('CAC', 'CAA', 'C', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AAT', 'AAG', 'T', 'G'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AAG', 'AAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('CAG', 'CAC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'TTG', 'A', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [[('AAG', 'AAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AAC', 'AAA', 'C', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAT', 'GAG', 'T', 'G'), 0]],\n",
       " [[('ATG', 'ATT', 'G', 'T'), 0], [('ATG', 'ATC', 'G', 'C'), 0]],\n",
       " [[('AAA', 'AAC', 'A', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGA', 'CGA', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'ATT', 'G', 'T'), 0], [('ATG', 'ATC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GTG', 'TTG', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TTC', 'TTG', 'C', 'G'), 0], [('TTC', 'CTC', 'T', 'C'), -2]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AAG', 'AAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AAC', 'AAA', 'C', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AGA', 'AGC', 'A', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAG', 'GAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAG', 'GAC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GTG', 'TTG', 'G', 'T'), 0]],\n",
       " [],\n",
       " [[('TTC', 'TTG', 'C', 'G'), 0], [('TTC', 'CTC', 'T', 'C'), -2]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGG', 'CGG', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GTG', 'CTG', 'G', 'C'), 0]],\n",
       " [],\n",
       " [[('ATG', 'ATT', 'G', 'T'), 0], [('ATG', 'ATA', 'G', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('CAG', 'CAC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGG', 'AGG', 'G', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAC', 'GAA', 'C', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'ATT', 'G', 'T'), 0], [('ATG', 'ATC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('CAG', 'CAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [[('GAC', 'GAA', 'C', 'A'), 0]],\n",
       " [[('GAC', 'GAA', 'C', 'A'), 0]],\n",
       " [[('CAG', 'CAC', 'G', 'C'), 0]],\n",
       " [[('AAC', 'AAA', 'C', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TTC', 'TTA', 'C', 'A'), 0], [('TTC', 'CTC', 'T', 'C'), -2]],\n",
       " [],\n",
       " [],\n",
       " [[('AAC', 'AAG', 'C', 'G'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('CAG', 'CAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGG', 'CGG', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TTG', 'TTT', 'G', 'T'), 0]],\n",
       " [[('TTT', 'TTA', 'T', 'A'), 2], [('TTT', 'TTG', 'T', 'G'), 2]],\n",
       " [[('TGG', 'AGG', 'T', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('CAG', 'CAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [[('GGA', 'CGA', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'ATC', 'G', 'C'), 0], [('ATG', 'ATA', 'G', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAC', 'GAG', 'C', 'G'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'CTG', 'A', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GTG', 'TTG', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'CTG', 'A', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAC', 'GAG', 'C', 'G'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TGG', 'AGG', 'T', 'A'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GTG', 'TTG', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AAT', 'AAG', 'T', 'G'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('CAG', 'CAC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGG', 'AGG', 'G', 'A'), 0]],\n",
       " [[('AGG', 'AGC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TTC', 'TTA', 'C', 'A'), 0], [('TTC', 'CTC', 'T', 'C'), -2]],\n",
       " [],\n",
       " [[('CAG', 'CAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GTG', 'TTG', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [[('GAC', 'GAG', 'C', 'G'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'ATT', 'G', 'T'), 0], [('ATG', 'ATC', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('TTG', 'TTT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'ATT', 'G', 'T'), 0], [('ATG', 'ATA', 'G', 'A'), 0]],\n",
       " [],\n",
       " [[('AGG', 'AGT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [[('TGC', 'TCC', 'G', 'C'), 1]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ACC', 'TCC', 'A', 'T'), -1]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAG', 'GAT', 'G', 'T'), 0]],\n",
       " [[('TTT', 'TTA', 'T', 'A'), 2], [('TTT', 'TTG', 'T', 'G'), 2]],\n",
       " [[('GAG', 'GAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AAG', 'AAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GGA', 'CGA', 'G', 'C'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('ATG', 'ATT', 'G', 'T'), 0], [('ATG', 'ATC', 'G', 'C'), 0]],\n",
       " [[('ACT', 'TCT', 'A', 'T'), -1]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('GAC', 'GAG', 'C', 'G'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " [[('AAG', 'AAT', 'G', 'T'), 0]],\n",
       " [],\n",
       " [],\n",
       " [],\n",
       " ...]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "additional_amino_list = compare_lists(ref_codon_list, alt_amino_list, alt_codon_list)\n",
    "additional_amino_list"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a076c48",
   "metadata": {},
   "source": [
    "From the *additional_amino_list*, create an *additional_amino_df* dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "656ad1b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432375</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432376</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432377</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432378</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432379</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>432380 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           0     1\n",
       "0       None  None\n",
       "1       None  None\n",
       "2       None  None\n",
       "3       None  None\n",
       "4       None  None\n",
       "...      ...   ...\n",
       "432375  None  None\n",
       "432376  None  None\n",
       "432377  None  None\n",
       "432378  None  None\n",
       "432379  None  None\n",
       "\n",
       "[432380 rows x 2 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "additional_amino_df = pd.DataFrame(additional_amino_list)\n",
    "additional_amino_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81fa1524",
   "metadata": {},
   "source": [
    "Join columns from *df_2* (#Uploaded_variation, Allele, Gene, cDNA_position, CDS_position, Protein_position, Amino_acids, Codons; clinvar_vep_ref.vcf, missense_variant, Canonical) and *additional_amino_df*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "2ae9e99f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>711716</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>3162</td>\n",
       "      <td>3092</td>\n",
       "      <td>1031</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGa/cAa</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18171</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>3068</td>\n",
       "      <td>2998</td>\n",
       "      <td>1000</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>18172</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>2985</td>\n",
       "      <td>2915</td>\n",
       "      <td>972</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>tGt/tAt</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>18174</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>2181</td>\n",
       "      <td>2111</td>\n",
       "      <td>704</td>\n",
       "      <td>R/H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>729811</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>899</td>\n",
       "      <td>829</td>\n",
       "      <td>277</td>\n",
       "      <td>D/N</td>\n",
       "      <td>Gac/Aac</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432375</th>\n",
       "      <td>155772</td>\n",
       "      <td>T</td>\n",
       "      <td>57688</td>\n",
       "      <td>3502</td>\n",
       "      <td>3487</td>\n",
       "      <td>1163</td>\n",
       "      <td>R/W</td>\n",
       "      <td>Cgg/Tgg</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432376</th>\n",
       "      <td>450504</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>3506</td>\n",
       "      <td>3491</td>\n",
       "      <td>1164</td>\n",
       "      <td>H/R</td>\n",
       "      <td>cAc/cGc</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432377</th>\n",
       "      <td>1424334</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>3511</td>\n",
       "      <td>3496</td>\n",
       "      <td>1166</td>\n",
       "      <td>S/G</td>\n",
       "      <td>Agt/Ggt</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432378</th>\n",
       "      <td>1417067</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>3547</td>\n",
       "      <td>3532</td>\n",
       "      <td>1178</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Acc/Gcc</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432379</th>\n",
       "      <td>1353194</td>\n",
       "      <td>T</td>\n",
       "      <td>57688</td>\n",
       "      <td>3584</td>\n",
       "      <td>3569</td>\n",
       "      <td>1190</td>\n",
       "      <td>T/I</td>\n",
       "      <td>aCa/aTa</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>432380 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #Uploaded_variation Allele   Gene cDNA_position CDS_position  \\\n",
       "0                    711716      T      2          3162         3092   \n",
       "1                     18171      C      2          3068         2998   \n",
       "2                     18172      T      2          2985         2915   \n",
       "3                     18174      T      2          2181         2111   \n",
       "4                    729811      T      2           899          829   \n",
       "...                     ...    ...    ...           ...          ...   \n",
       "432375               155772      T  57688          3502         3487   \n",
       "432376               450504      G  57688          3506         3491   \n",
       "432377              1424334      G  57688          3511         3496   \n",
       "432378              1417067      G  57688          3547         3532   \n",
       "432379              1353194      T  57688          3584         3569   \n",
       "\n",
       "       Protein_position Amino_acids   Codons     0     1  \n",
       "0                  1031         R/Q  cGa/cAa  None  None  \n",
       "1                  1000         I/V  Atc/Gtc  None  None  \n",
       "2                   972         C/Y  tGt/tAt  None  None  \n",
       "3                   704         R/H  cGt/cAt  None  None  \n",
       "4                   277         D/N  Gac/Aac  None  None  \n",
       "...                 ...         ...      ...   ...   ...  \n",
       "432375             1163         R/W  Cgg/Tgg  None  None  \n",
       "432376             1164         H/R  cAc/cGc  None  None  \n",
       "432377             1166         S/G  Agt/Ggt  None  None  \n",
       "432378             1178         T/A  Acc/Gcc  None  None  \n",
       "432379             1190         T/I  aCa/aTa  None  None  \n",
       "\n",
       "[432380 rows x 10 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_3 = pd.concat([df_2,additional_amino_df], axis=1)\n",
    "df_3"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cc24403e",
   "metadata": {},
   "source": [
    "Create a list of required columns."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "20aecc12",
   "metadata": {},
   "outputs": [],
   "source": [
    "ncol = len(df_3.columns)\n",
    "col_list = [0]\n",
    "for i in range(6, ncol):\n",
    "    col_list.append(i)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2396904e",
   "metadata": {},
   "source": [
    "Get the required columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "3953e4ad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>711716</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGa/cAa</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18171</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>18172</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>tGt/tAt</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>18174</td>\n",
       "      <td>R/H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>729811</td>\n",
       "      <td>D/N</td>\n",
       "      <td>Gac/Aac</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432375</th>\n",
       "      <td>155772</td>\n",
       "      <td>R/W</td>\n",
       "      <td>Cgg/Tgg</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432376</th>\n",
       "      <td>450504</td>\n",
       "      <td>H/R</td>\n",
       "      <td>cAc/cGc</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432377</th>\n",
       "      <td>1424334</td>\n",
       "      <td>S/G</td>\n",
       "      <td>Agt/Ggt</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432378</th>\n",
       "      <td>1417067</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Acc/Gcc</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432379</th>\n",
       "      <td>1353194</td>\n",
       "      <td>T/I</td>\n",
       "      <td>aCa/aTa</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>432380 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        #Uploaded_variation Amino_acids   Codons     0     1\n",
       "0                    711716         R/Q  cGa/cAa  None  None\n",
       "1                     18171         I/V  Atc/Gtc  None  None\n",
       "2                     18172         C/Y  tGt/tAt  None  None\n",
       "3                     18174         R/H  cGt/cAt  None  None\n",
       "4                    729811         D/N  Gac/Aac  None  None\n",
       "...                     ...         ...      ...   ...   ...\n",
       "432375               155772         R/W  Cgg/Tgg  None  None\n",
       "432376               450504         H/R  cAc/cGc  None  None\n",
       "432377              1424334         S/G  Agt/Ggt  None  None\n",
       "432378              1417067         T/A  Acc/Gcc  None  None\n",
       "432379              1353194         T/I  aCa/aTa  None  None\n",
       "\n",
       "[432380 rows x 5 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_4 = df_3.iloc[:, col_list]\n",
    "df_4"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3aab9c2f",
   "metadata": {},
   "source": [
    "Get the strings containing new alternative codons only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "ef10dd56",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\Miniconda3\\lib\\site-packages\\pandas\\core\\frame.py:5039: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  return super().rename(\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>372642</td>\n",
       "      <td>E/D</td>\n",
       "      <td>gaG/gaT</td>\n",
       "      <td>[(GAG, GAC, G, C), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>800252</td>\n",
       "      <td>V/L</td>\n",
       "      <td>Gtg/Ttg</td>\n",
       "      <td>[(GTG, CTG, G, C), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>644546</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>caG/caT</td>\n",
       "      <td>[(CAG, CAC, G, C), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>1515743</td>\n",
       "      <td>M/I</td>\n",
       "      <td>atG/atC</td>\n",
       "      <td>[(ATG, ATT, G, T), 0]</td>\n",
       "      <td>[(ATG, ATA, G, A), 0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>561776</td>\n",
       "      <td>S/R</td>\n",
       "      <td>agT/agG</td>\n",
       "      <td>[(AGT, CGT, A, C), -2]</td>\n",
       "      <td>[(AGT, AGA, T, A), 0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>1438772</td>\n",
       "      <td>M/L</td>\n",
       "      <td>Atg/Ctg</td>\n",
       "      <td>[(ATG, TTG, A, T), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>1043900</td>\n",
       "      <td>V/L</td>\n",
       "      <td>Gtg/Ctg</td>\n",
       "      <td>[(GTG, TTG, G, T), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>860781</td>\n",
       "      <td>W/R</td>\n",
       "      <td>Tgg/Cgg</td>\n",
       "      <td>[(TGG, AGG, T, A), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>120256</td>\n",
       "      <td>D/E</td>\n",
       "      <td>gaC/gaA</td>\n",
       "      <td>[(GAC, GAG, C, G), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>1403638</td>\n",
       "      <td>G/R</td>\n",
       "      <td>Gga/Aga</td>\n",
       "      <td>[(GGA, CGA, G, C), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>942534</td>\n",
       "      <td>G/R</td>\n",
       "      <td>Gga/Aga</td>\n",
       "      <td>[(GGA, CGA, G, C), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>963431</td>\n",
       "      <td>G/R</td>\n",
       "      <td>Ggg/Agg</td>\n",
       "      <td>[(GGG, CGG, G, C), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>1019847</td>\n",
       "      <td>S/R</td>\n",
       "      <td>agC/agG</td>\n",
       "      <td>[(AGC, CGC, A, C), -2]</td>\n",
       "      <td>[(AGC, AGA, C, A), 0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>546698</td>\n",
       "      <td>F/L</td>\n",
       "      <td>Ttt/Ctt</td>\n",
       "      <td>[(TTT, TTA, T, A), 2]</td>\n",
       "      <td>[(TTT, TTG, T, G), 2]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>939899</td>\n",
       "      <td>W/C</td>\n",
       "      <td>tgG/tgC</td>\n",
       "      <td>[(TGG, TGT, G, T), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>1008086</td>\n",
       "      <td>N/K</td>\n",
       "      <td>aaT/aaA</td>\n",
       "      <td>[(AAT, AAG, T, G), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>451644</td>\n",
       "      <td>M/L</td>\n",
       "      <td>Atg/Ttg</td>\n",
       "      <td>[(ATG, CTG, A, C), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>617557</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>caA/caC</td>\n",
       "      <td>[(CAA, CAT, A, T), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>1465244</td>\n",
       "      <td>E/D</td>\n",
       "      <td>gaG/gaC</td>\n",
       "      <td>[(GAG, GAT, G, T), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>1403325</td>\n",
       "      <td>M/I</td>\n",
       "      <td>atG/atA</td>\n",
       "      <td>[(ATG, ATT, G, T), 0]</td>\n",
       "      <td>[(ATG, ATC, G, C), 0]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          ID Amino_acids   Codons                       0  \\\n",
       "24    372642         E/D  gaG/gaT   [(GAG, GAC, G, C), 0]   \n",
       "30    800252         V/L  Gtg/Ttg   [(GTG, CTG, G, C), 0]   \n",
       "38    644546         Q/H  caG/caT   [(CAG, CAC, G, C), 0]   \n",
       "51   1515743         M/I  atG/atC   [(ATG, ATT, G, T), 0]   \n",
       "56    561776         S/R  agT/agG  [(AGT, CGT, A, C), -2]   \n",
       "65   1438772         M/L  Atg/Ctg   [(ATG, TTG, A, T), 0]   \n",
       "67   1043900         V/L  Gtg/Ctg   [(GTG, TTG, G, T), 0]   \n",
       "70    860781         W/R  Tgg/Cgg   [(TGG, AGG, T, A), 0]   \n",
       "76    120256         D/E  gaC/gaA   [(GAC, GAG, C, G), 0]   \n",
       "78   1403638         G/R  Gga/Aga   [(GGA, CGA, G, C), 0]   \n",
       "82    942534         G/R  Gga/Aga   [(GGA, CGA, G, C), 0]   \n",
       "87    963431         G/R  Ggg/Agg   [(GGG, CGG, G, C), 0]   \n",
       "91   1019847         S/R  agC/agG  [(AGC, CGC, A, C), -2]   \n",
       "100   546698         F/L  Ttt/Ctt   [(TTT, TTA, T, A), 2]   \n",
       "103   939899         W/C  tgG/tgC   [(TGG, TGT, G, T), 0]   \n",
       "104  1008086         N/K  aaT/aaA   [(AAT, AAG, T, G), 0]   \n",
       "107   451644         M/L  Atg/Ttg   [(ATG, CTG, A, C), 0]   \n",
       "112   617557         Q/H  caA/caC   [(CAA, CAT, A, T), 0]   \n",
       "146  1465244         E/D  gaG/gaC   [(GAG, GAT, G, T), 0]   \n",
       "150  1403325         M/I  atG/atA   [(ATG, ATT, G, T), 0]   \n",
       "\n",
       "                         1  \n",
       "24                    None  \n",
       "30                    None  \n",
       "38                    None  \n",
       "51   [(ATG, ATA, G, A), 0]  \n",
       "56   [(AGT, AGA, T, A), 0]  \n",
       "65                    None  \n",
       "67                    None  \n",
       "70                    None  \n",
       "76                    None  \n",
       "78                    None  \n",
       "82                    None  \n",
       "87                    None  \n",
       "91   [(AGC, AGA, C, A), 0]  \n",
       "100  [(TTT, TTG, T, G), 2]  \n",
       "103                   None  \n",
       "104                   None  \n",
       "107                   None  \n",
       "112                   None  \n",
       "146                   None  \n",
       "150  [(ATG, ATC, G, C), 0]  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_5 = df_4.loc[df_4[0].isna()!=True]\n",
    "df_5.rename(columns={'#Uploaded_variation': 'ID'}, inplace=True)\n",
    "df_5.head(20)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f1ef174",
   "metadata": {},
   "source": [
    "## 4. Creating VCF "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "217566fa",
   "metadata": {},
   "source": [
    "Merge df_5 and df_clinvar_2 (CHROM, POS, ID, REF, ALT_1, CLNSIG из clinvar.vcf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "8f83c5d0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\1\\AppData\\Local\\Temp/ipykernel_19652/2067860276.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_clinvar_2['ID']=df_clinvar_2['ID'].astype(int)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12</td>\n",
       "      <td>8823290</td>\n",
       "      <td>372642</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>E/D</td>\n",
       "      <td>gaG/gaT</td>\n",
       "      <td>[(GAG, GAC, G, C), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12</td>\n",
       "      <td>8823750</td>\n",
       "      <td>800252</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>V/L</td>\n",
       "      <td>Gtg/Ttg</td>\n",
       "      <td>[(GTG, CTG, G, C), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12</td>\n",
       "      <td>8823848</td>\n",
       "      <td>644546</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>Q/H</td>\n",
       "      <td>caG/caT</td>\n",
       "      <td>[(CAG, CAC, G, C), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12</td>\n",
       "      <td>8835599</td>\n",
       "      <td>1515743</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>M/I</td>\n",
       "      <td>atG/atC</td>\n",
       "      <td>[(ATG, ATT, G, T), 0]</td>\n",
       "      <td>[(ATG, ATA, G, A), 0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12</td>\n",
       "      <td>8835653</td>\n",
       "      <td>561776</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>S/R</td>\n",
       "      <td>agT/agG</td>\n",
       "      <td>[(AGT, CGT, A, C), -2]</td>\n",
       "      <td>[(AGT, AGA, T, A), 0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53846</th>\n",
       "      <td>5</td>\n",
       "      <td>61521384</td>\n",
       "      <td>1381845</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>H/Q</td>\n",
       "      <td>caT/caG</td>\n",
       "      <td>[(CAT, CAA, T, A), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53847</th>\n",
       "      <td>5</td>\n",
       "      <td>61538904</td>\n",
       "      <td>1311009</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>W/C</td>\n",
       "      <td>tgG/tgC</td>\n",
       "      <td>[(TGG, TGT, G, T), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53848</th>\n",
       "      <td>5</td>\n",
       "      <td>61539627</td>\n",
       "      <td>1347650</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>E/D</td>\n",
       "      <td>gaA/gaC</td>\n",
       "      <td>[(GAA, GAT, A, T), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53849</th>\n",
       "      <td>5</td>\n",
       "      <td>61543813</td>\n",
       "      <td>1514466</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>K/N</td>\n",
       "      <td>aaG/aaT</td>\n",
       "      <td>[(AAG, AAC, G, C), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53850</th>\n",
       "      <td>5</td>\n",
       "      <td>61543957</td>\n",
       "      <td>1490854</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>K/N</td>\n",
       "      <td>aaG/aaC</td>\n",
       "      <td>[(AAG, AAT, G, T), 0]</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53851 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM       POS       ID REF ALT_1                  CLNSIG Amino_acids  \\\n",
       "0        12   8823290   372642   G     T  Uncertain_significance         E/D   \n",
       "1        12   8823750   800252   G     T           Likely_benign         V/L   \n",
       "2        12   8823848   644546   G     T  Uncertain_significance         Q/H   \n",
       "3        12   8835599  1515743   G     C  Uncertain_significance         M/I   \n",
       "4        12   8835653   561776   T     G  Uncertain_significance         S/R   \n",
       "...     ...       ...      ...  ..   ...                     ...         ...   \n",
       "53846     5  61521384  1381845   T     G  Uncertain_significance         H/Q   \n",
       "53847     5  61538904  1311009   G     C  Uncertain_significance         W/C   \n",
       "53848     5  61539627  1347650   A     C  Uncertain_significance         E/D   \n",
       "53849     5  61543813  1514466   G     T  Uncertain_significance         K/N   \n",
       "53850     5  61543957  1490854   G     C  Uncertain_significance         K/N   \n",
       "\n",
       "        Codons                       0                      1  \n",
       "0      gaG/gaT   [(GAG, GAC, G, C), 0]                   None  \n",
       "1      Gtg/Ttg   [(GTG, CTG, G, C), 0]                   None  \n",
       "2      caG/caT   [(CAG, CAC, G, C), 0]                   None  \n",
       "3      atG/atC   [(ATG, ATT, G, T), 0]  [(ATG, ATA, G, A), 0]  \n",
       "4      agT/agG  [(AGT, CGT, A, C), -2]  [(AGT, AGA, T, A), 0]  \n",
       "...        ...                     ...                    ...  \n",
       "53846  caT/caG   [(CAT, CAA, T, A), 0]                   None  \n",
       "53847  tgG/tgC   [(TGG, TGT, G, T), 0]                   None  \n",
       "53848  gaA/gaC   [(GAA, GAT, A, T), 0]                   None  \n",
       "53849  aaG/aaT   [(AAG, AAC, G, C), 0]                   None  \n",
       "53850  aaG/aaC   [(AAG, AAT, G, T), 0]                   None  \n",
       "\n",
       "[53851 rows x 10 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_clinvar_2 = df_clinvar.iloc[:, [0,1,2,3,4,18]]\n",
    "df_clinvar_2['ID']=df_clinvar_2['ID'].astype(int)\n",
    "df_add = df_clinvar_2.merge(df_5, on='ID', how='right')\n",
    "df_add"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "79596039",
   "metadata": {},
   "source": [
    "Form the VCF format. Create the function that creates a preliminary list for the dataframe. This list contains columns for VCF format: #CHROM, POS, ID, REF, ALT, QUAL, FILTER, INFO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "e40ccb0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_df_list(df):\n",
    "    \"\"\"\n",
    "    creates a list to create a dataframe \n",
    "        :param df: df_add\n",
    "    :return df_list: list containing information for vcf formate\n",
    "    \"\"\"\n",
    "\n",
    "\n",
    "    df_list=[]\n",
    "    for row in df.itertuples():        \n",
    "        for i in range(9, len(row)): \n",
    "            if row[i] and row[i] != '#':\n",
    "                str_list = []\n",
    "                pos = int(row[2]) + int(row[i][1])\n",
    "                ID = str(row[3]) + '_' + str(i-8)\n",
    "                REF = row[i][0][2]\n",
    "                ALT_1 = row[i][0][3]\n",
    "                REF_codon = row[i][0][0]\n",
    "                ALT_codon = row[i][0][1]\n",
    "                str_list.append(row[1])\n",
    "                str_list.append(pos)\n",
    "                str_list.append(ID)\n",
    "                str_list.append(REF)\n",
    "                str_list.append(ALT_1)\n",
    "                str_list.append('.') # 'QUAL'\n",
    "                str_list.append('.') # 'FILTER'\n",
    "                info_str = ''\n",
    "                for k in range(6, 8):\n",
    "                    info_str += str(row[k]) + ';'\n",
    "                info_str += REF_codon + ';' + ALT_codon\n",
    "                str_list.append(info_str)\n",
    "                df_list.append(str_list)\n",
    "    return df_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "e49d42fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_list = make_df_list(df_add)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4dc8cc77",
   "metadata": {},
   "source": [
    "Create a dataframe *df_6* from the list *df_list*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "d1429a98",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#CHROM</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>8823290</td>\n",
       "      <td>372642_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain_significance;E/D;GAG;GAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>8823750</td>\n",
       "      <td>800252_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_benign;V/L;GTG;CTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>8823848</td>\n",
       "      <td>644546_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain_significance;Q/H;CAG;CAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>8835599</td>\n",
       "      <td>1515743_1</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain_significance;M/I;ATG;ATT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>8835599</td>\n",
       "      <td>1515743_2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain_significance;M/I;ATG;ATA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>61521384</td>\n",
       "      <td>1381845_1</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain_significance;H/Q;CAT;CAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>61538904</td>\n",
       "      <td>1311009_1</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain_significance;W/C;TGG;TGT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>61539627</td>\n",
       "      <td>1347650_1</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain_significance;E/D;GAA;GAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>61543813</td>\n",
       "      <td>1514466_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain_significance;K/N;AAG;AAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>61543957</td>\n",
       "      <td>1490854_1</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Uncertain_significance;K/N;AAG;AAT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>63383 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             POS         ID REF ALT QUAL FILTER  \\\n",
       "#CHROM                                            \n",
       "12       8823290   372642_1   G   C    .      .   \n",
       "12       8823750   800252_1   G   C    .      .   \n",
       "12       8823848   644546_1   G   C    .      .   \n",
       "12       8835599  1515743_1   G   T    .      .   \n",
       "12       8835599  1515743_2   G   A    .      .   \n",
       "...          ...        ...  ..  ..  ...    ...   \n",
       "5       61521384  1381845_1   T   A    .      .   \n",
       "5       61538904  1311009_1   G   T    .      .   \n",
       "5       61539627  1347650_1   A   T    .      .   \n",
       "5       61543813  1514466_1   G   C    .      .   \n",
       "5       61543957  1490854_1   G   T    .      .   \n",
       "\n",
       "                                      INFO  \n",
       "#CHROM                                      \n",
       "12      Uncertain_significance;E/D;GAG;GAC  \n",
       "12               Likely_benign;V/L;GTG;CTG  \n",
       "12      Uncertain_significance;Q/H;CAG;CAC  \n",
       "12      Uncertain_significance;M/I;ATG;ATT  \n",
       "12      Uncertain_significance;M/I;ATG;ATA  \n",
       "...                                    ...  \n",
       "5       Uncertain_significance;H/Q;CAT;CAA  \n",
       "5       Uncertain_significance;W/C;TGG;TGT  \n",
       "5       Uncertain_significance;E/D;GAA;GAT  \n",
       "5       Uncertain_significance;K/N;AAG;AAC  \n",
       "5       Uncertain_significance;K/N;AAG;AAT  \n",
       "\n",
       "[63383 rows x 7 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_columns = ['#CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO']\n",
    "df_6 = pd.DataFrame(df_list, columns=df_columns)\n",
    "df_6.set_index('#CHROM', inplace=True)\n",
    "df_6"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "679da92c",
   "metadata": {},
   "source": [
    "Сreate the function to save the dataframe in vcf format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "e7113fb5",
   "metadata": {},
   "outputs": [],
   "source": [
    "def save_VCF(df, name):\n",
    "    \"\"\"\n",
    "    save dataframe to vcf file \n",
    "        :param df: dataframe in the vcf formate\n",
    "    \"\"\"\n",
    "    \n",
    "    \n",
    "    header = \"\"\"##fileformat=\n",
    "##fileDate=\n",
    "##source=\n",
    "##reference=\n",
    "#CHROM POS ID REF ALT QUAL FILTER INFO\n",
    "\"\"\"\n",
    "    with open(name, 'w') as vcf:\n",
    "        vcf.write(header)\n",
    "    df.to_csv(name, sep=\"\\t\", mode='a', header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ef02caf",
   "metadata": {},
   "source": [
    "## 5. Create 4 dataframes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "502ccf25",
   "metadata": {},
   "source": [
    "Create 4 dataframes from *df_26*:\n",
    ">benign_df:\n",
    "* Benign\n",
    "* Benign|_association\n",
    "* Benign|_association|_confers_sensitivity\n",
    "* Benign|_confers_sensitivity\n",
    "* Benign|_drug_response\n",
    "* Benign|_other\n",
    "* Benign|_protective\n",
    "* Benign|_risk_factor\n",
    "\n",
    ">likely_benign_df:\n",
    "* Benign/Likely_benign\n",
    "* Benign/Likely_benign|_drug_response\n",
    "* Benign/Likely_benign|_drug_response|_other\n",
    "* Benign/Likely_benign|_other\n",
    "* Benign/Likely_benign|_other|_risk_factor\n",
    "* Benign/Likely_benign|_risk_factor\n",
    "* Likely_benign|_drug_response|_other\n",
    "* Likely_benign|_other)\n",
    "\n",
    ">pathogenic:\n",
    "\n",
    "* pathogenic_df(Pathogenic\n",
    "* Pathogenic|_Affects\n",
    "* Pathogenic|_association\n",
    "* Pathogenic|_association|_protective\n",
    "* Pathogenic|_confers_sensitivity\n",
    "* Pathogenic|_drug_response\n",
    "* Pathogenic|_drug_response|_other\n",
    "* Pathogenic|_other\n",
    "* Pathogenic|_protective\n",
    "* Pathogenic|_risk_factor\n",
    "\n",
    ">likely_pathogenic_df:\n",
    "* Pathogenic/Likely_pathogenic\n",
    "* Pathogenic/Likely_pathogenic|_drug_response\n",
    "* Pathogenic/Likely_pathogenic|_other\n",
    "* Pathogenic/Likely_pathogenic|_risk_factor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "df23f0aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#CHROM</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>8838341</td>\n",
       "      <td>120256_1</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign;D/E;GAC;GAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>8849672</td>\n",
       "      <td>561635_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign;V/L;GTG;CTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>8852296</td>\n",
       "      <td>1169145_1</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign;D/E;GAC;GAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>8867913</td>\n",
       "      <td>413827_1</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign;N/K;AAC;AAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>44310332</td>\n",
       "      <td>136225_1</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign;D/E;GAC;GAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>88434626</td>\n",
       "      <td>320967_1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign;G/R;GGG;AGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>151286637</td>\n",
       "      <td>779222_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign;G/R;GGG;CGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>151289257</td>\n",
       "      <td>717811_1</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign;Q/H;CAA;CAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>76425072</td>\n",
       "      <td>791452_1</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign;E/D;GAG;GAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>76441894</td>\n",
       "      <td>712658_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign;R/S;AGG;AGC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1510 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              POS         ID REF ALT QUAL FILTER                INFO\n",
       "#CHROM                                                              \n",
       "12        8838341   120256_1   C   G    .      .  Benign;D/E;GAC;GAG\n",
       "12        8849672   561635_1   G   C    .      .  Benign;V/L;GTG;CTG\n",
       "12        8852296  1169145_1   C   G    .      .  Benign;D/E;GAC;GAG\n",
       "12        8867913   413827_1   C   A    .      .  Benign;N/K;AAC;AAA\n",
       "6        44310332   136225_1   C   G    .      .  Benign;D/E;GAC;GAG\n",
       "...           ...        ...  ..  ..  ...    ...                 ...\n",
       "16       88434626   320967_1   G   A    .      .  Benign;G/R;GGG;AGG\n",
       "1       151286637   779222_1   G   C    .      .  Benign;G/R;GGG;CGG\n",
       "1       151289257   717811_1   A   T    .      .  Benign;Q/H;CAA;CAT\n",
       "7        76425072   791452_1   G   T    .      .  Benign;E/D;GAG;GAT\n",
       "7        76441894   712658_1   G   C    .      .  Benign;R/S;AGG;AGC\n",
       "\n",
       "[1510 rows x 7 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "benign_df = df_6[(df_6['INFO'].str.contains(\"Benign\")==True) & (df_6['INFO'].str.contains(\"Likely_benign\")==False)]\n",
    "benign_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "7305c9b2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#CHROM</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>8823750</td>\n",
       "      <td>800252_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_benign;V/L;GTG;CTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>8868027</td>\n",
       "      <td>241909_1</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign/Likely_benign;E/D;GAG;GAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>53307533</td>\n",
       "      <td>309718_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign/Likely_benign;G/R;GGG;CGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>44300553</td>\n",
       "      <td>1201845_1</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_benign;Q/H;CAG;CAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>44306988</td>\n",
       "      <td>357073_1</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_benign;M/L;ATG;CTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>88436510</td>\n",
       "      <td>808133_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_benign;G/R;GGA;CGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>88437796</td>\n",
       "      <td>511466_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_benign;R/S;AGG;AGC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>88437797</td>\n",
       "      <td>388236_1</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_benign;G/R;GGG;AGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>88438445</td>\n",
       "      <td>195113_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Benign/Likely_benign;G/R;GGG;CGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>88438876</td>\n",
       "      <td>392227_1</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_benign;H/Q;CAC;CAG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2367 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             POS         ID REF ALT QUAL FILTER  \\\n",
       "#CHROM                                            \n",
       "12       8823750   800252_1   G   C    .      .   \n",
       "12       8868027   241909_1   G   T    .      .   \n",
       "12      53307533   309718_1   G   C    .      .   \n",
       "6       44300553  1201845_1   G   T    .      .   \n",
       "6       44306988   357073_1   A   C    .      .   \n",
       "...          ...        ...  ..  ..  ...    ...   \n",
       "16      88436510   808133_1   G   C    .      .   \n",
       "16      88437796   511466_1   G   C    .      .   \n",
       "16      88437797   388236_1   G   A    .      .   \n",
       "16      88438445   195113_1   G   C    .      .   \n",
       "16      88438876   392227_1   C   G    .      .   \n",
       "\n",
       "                                    INFO  \n",
       "#CHROM                                    \n",
       "12             Likely_benign;V/L;GTG;CTG  \n",
       "12      Benign/Likely_benign;E/D;GAG;GAT  \n",
       "12      Benign/Likely_benign;G/R;GGG;CGG  \n",
       "6              Likely_benign;Q/H;CAG;CAT  \n",
       "6              Likely_benign;M/L;ATG;CTG  \n",
       "...                                  ...  \n",
       "16             Likely_benign;G/R;GGA;CGA  \n",
       "16             Likely_benign;R/S;AGG;AGC  \n",
       "16             Likely_benign;G/R;GGG;AGG  \n",
       "16      Benign/Likely_benign;G/R;GGG;CGG  \n",
       "16             Likely_benign;H/Q;CAC;CAG  \n",
       "\n",
       "[2367 rows x 7 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "likely_benign_df = df_6[df_6['INFO'].str.contains(\"Likely_benign\")==True]\n",
    "likely_benign_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "ec97b38c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#CHROM</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>214990791</td>\n",
       "      <td>2863_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Pathogenic;G/R;GGA;CGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>94007725</td>\n",
       "      <td>99418_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Pathogenic;G/R;GGA;CGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>94010868</td>\n",
       "      <td>1184497_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Pathogenic;M/I;ATG;ATC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>94010868</td>\n",
       "      <td>1184497_2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Pathogenic;M/I;ATG;ATA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>94019708</td>\n",
       "      <td>1456885_1</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Pathogenic;M/I;ATG;ATT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>105801286</td>\n",
       "      <td>156583_1</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Pathogenic;S/R;AGT;CGT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>105801288</td>\n",
       "      <td>156583_2</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Pathogenic;S/R;AGT;AGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X</th>\n",
       "      <td>137567449</td>\n",
       "      <td>545553_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Pathogenic;C/S;TGC;TCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X</th>\n",
       "      <td>137567448</td>\n",
       "      <td>11437_1</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Pathogenic;C/S;TGC;AGC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>40285988</td>\n",
       "      <td>4272_1</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Pathogenic;W/R;TGG;AGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2420 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              POS         ID REF ALT QUAL FILTER                    INFO\n",
       "#CHROM                                                                  \n",
       "2       214990791     2863_1   G   C    .      .  Pathogenic;G/R;GGA;CGA\n",
       "1        94007725    99418_1   G   C    .      .  Pathogenic;G/R;GGA;CGA\n",
       "1        94010868  1184497_1   G   C    .      .  Pathogenic;M/I;ATG;ATC\n",
       "1        94010868  1184497_2   G   A    .      .  Pathogenic;M/I;ATG;ATA\n",
       "1        94019708  1456885_1   G   T    .      .  Pathogenic;M/I;ATG;ATT\n",
       "...           ...        ...  ..  ..  ...    ...                     ...\n",
       "8       105801286   156583_1   A   C    .      .  Pathogenic;S/R;AGT;CGT\n",
       "8       105801288   156583_2   T   G    .      .  Pathogenic;S/R;AGT;AGG\n",
       "X       137567449   545553_1   G   C    .      .  Pathogenic;C/S;TGC;TCC\n",
       "X       137567448    11437_1   T   A    .      .  Pathogenic;C/S;TGC;AGC\n",
       "1        40285988     4272_1   T   A    .      .  Pathogenic;W/R;TGG;AGG\n",
       "\n",
       "[2420 rows x 7 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic_df = df_6[(df_6['INFO'].str.contains(\"Pathogenic\")==True) & (df_6['INFO'].str.contains(\"Likely_pathogenic\")==False)]\n",
    "pathogenic_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "b3adfbb9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#CHROM</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>70276973</td>\n",
       "      <td>549673_1</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_pathogenic;F/L;TTT;TTA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>70276973</td>\n",
       "      <td>549673_2</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_pathogenic;F/L;TTT;TTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>104785411</td>\n",
       "      <td>1323811_1</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_pathogenic;Q/H;CAG;CAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2286744</td>\n",
       "      <td>932907_1</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_pathogenic;N/K;AAC;AAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2288150</td>\n",
       "      <td>932908_1</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_pathogenic;L/F;TTG;TTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>32755717</td>\n",
       "      <td>1056_2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Pathogenic/Likely_pathogenic;F/L;TTC;CTC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>76329934</td>\n",
       "      <td>438806_1</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_pathogenic;D/E;GAC;GAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>144389856</td>\n",
       "      <td>290241_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_pathogenic;C/S;TGC;TCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>99985328</td>\n",
       "      <td>203372_1</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Likely_pathogenic;H/Q;CAT;CAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>45950360</td>\n",
       "      <td>998091_1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>Pathogenic/Likely_pathogenic;G/R;GGG;CGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2750 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              POS         ID REF ALT QUAL FILTER  \\\n",
       "#CHROM                                             \n",
       "16       70276973   549673_1   T   A    .      .   \n",
       "16       70276973   549673_2   T   G    .      .   \n",
       "9       104785411  1323811_1   G   T    .      .   \n",
       "16        2286744   932907_1   C   A    .      .   \n",
       "16        2288150   932908_1   G   T    .      .   \n",
       "...           ...        ...  ..  ..  ...    ...   \n",
       "12       32755717     1056_2   T   C    .      .   \n",
       "7        76329934   438806_1   C   A    .      .   \n",
       "2       144389856   290241_1   G   C    .      .   \n",
       "13       99985328   203372_1   T   A    .      .   \n",
       "20       45950360   998091_1   G   C    .      .   \n",
       "\n",
       "                                            INFO  \n",
       "#CHROM                                            \n",
       "16                 Likely_pathogenic;F/L;TTT;TTA  \n",
       "16                 Likely_pathogenic;F/L;TTT;TTG  \n",
       "9                  Likely_pathogenic;Q/H;CAG;CAT  \n",
       "16                 Likely_pathogenic;N/K;AAC;AAA  \n",
       "16                 Likely_pathogenic;L/F;TTG;TTT  \n",
       "...                                          ...  \n",
       "12      Pathogenic/Likely_pathogenic;F/L;TTC;CTC  \n",
       "7                  Likely_pathogenic;D/E;GAC;GAA  \n",
       "2                  Likely_pathogenic;C/S;TGC;TCC  \n",
       "13                 Likely_pathogenic;H/Q;CAT;CAA  \n",
       "20      Pathogenic/Likely_pathogenic;G/R;GGG;CGG  \n",
       "\n",
       "[2750 rows x 7 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "likely_pathogenic_df = df_6[df_6['INFO'].str.contains(\"Likely_pathogenic\")==True]\n",
    "likely_pathogenic_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "07388e52",
   "metadata": {},
   "source": [
    "Save dataframes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "ed664753",
   "metadata": {},
   "outputs": [],
   "source": [
    "save_VCF(benign_df, \"benign.vcf\")\n",
    "save_VCF(likely_benign_df, \"likely_benign.vcf\")\n",
    "save_VCF(pathogenic_df, \"pathogenic.vcf\")\n",
    "save_VCF(likely_pathogenic_df, \"likely_pathogenic.vcf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c24dec74",
   "metadata": {},
   "source": [
    "## 6. Descriptive statistics of the results obtained"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f6ed4b2a",
   "metadata": {},
   "source": [
    "Сount the number of clinically significant groups (Benign, Likely_benign, Pathogenic, Likely_pathogenic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "3f8f5aa5",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_1.rename(columns={'#Uploaded_variation': 'ID'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "5f040f62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT_1</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>Extra</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12</td>\n",
       "      <td>9079271</td>\n",
       "      <td>711716</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>12:9079271</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3162</td>\n",
       "      <td>3092</td>\n",
       "      <td>1031</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>cGa/cAa</td>\n",
       "      <td>rs202210880,COSV59389075</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12</td>\n",
       "      <td>9079672</td>\n",
       "      <td>18171</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Benign</td>\n",
       "      <td>12:9079672</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3068</td>\n",
       "      <td>2998</td>\n",
       "      <td>1000</td>\n",
       "      <td>I/V</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>rs669,CM980001,COSV59386644</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12</td>\n",
       "      <td>9079755</td>\n",
       "      <td>18172</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Benign</td>\n",
       "      <td>12:9079755</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2985</td>\n",
       "      <td>2915</td>\n",
       "      <td>972</td>\n",
       "      <td>C/Y</td>\n",
       "      <td>tGt/tAt</td>\n",
       "      <td>rs1800433,CM920001</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12</td>\n",
       "      <td>9094987</td>\n",
       "      <td>18174</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Benign</td>\n",
       "      <td>12:9094987</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2181</td>\n",
       "      <td>2111</td>\n",
       "      <td>704</td>\n",
       "      <td>R/H</td>\n",
       "      <td>cGt/cAt</td>\n",
       "      <td>rs1800434,COSV59390885,COSV59390935</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12</td>\n",
       "      <td>9107574</td>\n",
       "      <td>729811</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Benign</td>\n",
       "      <td>12:9107574</td>\n",
       "      <td>T</td>\n",
       "      <td>2</td>\n",
       "      <td>NM_000014.6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>899</td>\n",
       "      <td>829</td>\n",
       "      <td>277</td>\n",
       "      <td>D/N</td>\n",
       "      <td>Gac/Aac</td>\n",
       "      <td>rs140258425</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432375</th>\n",
       "      <td>5</td>\n",
       "      <td>61544156</td>\n",
       "      <td>155772</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>5:61544156</td>\n",
       "      <td>T</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3502</td>\n",
       "      <td>3487</td>\n",
       "      <td>1163</td>\n",
       "      <td>R/W</td>\n",
       "      <td>Cgg/Tgg</td>\n",
       "      <td>rs587777695,CM147511</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432376</th>\n",
       "      <td>5</td>\n",
       "      <td>61544160</td>\n",
       "      <td>450504</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>5:61544160</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3506</td>\n",
       "      <td>3491</td>\n",
       "      <td>1164</td>\n",
       "      <td>H/R</td>\n",
       "      <td>cAc/cGc</td>\n",
       "      <td>rs1554041457</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432377</th>\n",
       "      <td>5</td>\n",
       "      <td>61544165</td>\n",
       "      <td>1424334</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>5:61544165</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3511</td>\n",
       "      <td>3496</td>\n",
       "      <td>1166</td>\n",
       "      <td>S/G</td>\n",
       "      <td>Agt/Ggt</td>\n",
       "      <td>-</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432378</th>\n",
       "      <td>5</td>\n",
       "      <td>61544201</td>\n",
       "      <td>1417067</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>5:61544201</td>\n",
       "      <td>G</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3547</td>\n",
       "      <td>3532</td>\n",
       "      <td>1178</td>\n",
       "      <td>T/A</td>\n",
       "      <td>Acc/Gcc</td>\n",
       "      <td>rs1203405019</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432379</th>\n",
       "      <td>5</td>\n",
       "      <td>61544238</td>\n",
       "      <td>1353194</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>5:61544238</td>\n",
       "      <td>T</td>\n",
       "      <td>57688</td>\n",
       "      <td>NM_020928.2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3584</td>\n",
       "      <td>3569</td>\n",
       "      <td>1190</td>\n",
       "      <td>T/I</td>\n",
       "      <td>aCa/aTa</td>\n",
       "      <td>rs1436393669</td>\n",
       "      <td>IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>432380 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM       POS       ID REF ALT_1                        CLNSIG  \\\n",
       "0         12   9079271   711716   C     T                 Likely_benign   \n",
       "1         12   9079672    18171   T     C                        Benign   \n",
       "2         12   9079755    18172   C     T                        Benign   \n",
       "3         12   9094987    18174   C     T                        Benign   \n",
       "4         12   9107574   729811   C     T                        Benign   \n",
       "...      ...       ...      ...  ..   ...                           ...   \n",
       "432375     5  61544156   155772   C     T  Pathogenic/Likely_pathogenic   \n",
       "432376     5  61544160   450504   A     G             Likely_pathogenic   \n",
       "432377     5  61544165  1424334   A     G        Uncertain_significance   \n",
       "432378     5  61544201  1417067   A     G        Uncertain_significance   \n",
       "432379     5  61544238  1353194   C     T        Uncertain_significance   \n",
       "\n",
       "          Location Allele   Gene      Feature Feature_type       Consequence  \\\n",
       "0       12:9079271      T      2  NM_000014.6   Transcript  missense_variant   \n",
       "1       12:9079672      C      2  NM_000014.6   Transcript  missense_variant   \n",
       "2       12:9079755      T      2  NM_000014.6   Transcript  missense_variant   \n",
       "3       12:9094987      T      2  NM_000014.6   Transcript  missense_variant   \n",
       "4       12:9107574      T      2  NM_000014.6   Transcript  missense_variant   \n",
       "...            ...    ...    ...          ...          ...               ...   \n",
       "432375  5:61544156      T  57688  NM_020928.2   Transcript  missense_variant   \n",
       "432376  5:61544160      G  57688  NM_020928.2   Transcript  missense_variant   \n",
       "432377  5:61544165      G  57688  NM_020928.2   Transcript  missense_variant   \n",
       "432378  5:61544201      G  57688  NM_020928.2   Transcript  missense_variant   \n",
       "432379  5:61544238      T  57688  NM_020928.2   Transcript  missense_variant   \n",
       "\n",
       "       cDNA_position CDS_position Protein_position Amino_acids   Codons  \\\n",
       "0               3162         3092             1031         R/Q  cGa/cAa   \n",
       "1               3068         2998             1000         I/V  Atc/Gtc   \n",
       "2               2985         2915              972         C/Y  tGt/tAt   \n",
       "3               2181         2111              704         R/H  cGt/cAt   \n",
       "4                899          829              277         D/N  Gac/Aac   \n",
       "...              ...          ...              ...         ...      ...   \n",
       "432375          3502         3487             1163         R/W  Cgg/Tgg   \n",
       "432376          3506         3491             1164         H/R  cAc/cGc   \n",
       "432377          3511         3496             1166         S/G  Agt/Ggt   \n",
       "432378          3547         3532             1178         T/A  Acc/Gcc   \n",
       "432379          3584         3569             1190         T/I  aCa/aTa   \n",
       "\n",
       "                         Existing_variation  \\\n",
       "0                  rs202210880,COSV59389075   \n",
       "1               rs669,CM980001,COSV59386644   \n",
       "2                        rs1800433,CM920001   \n",
       "3       rs1800434,COSV59390885,COSV59390935   \n",
       "4                               rs140258425   \n",
       "...                                     ...   \n",
       "432375                 rs587777695,CM147511   \n",
       "432376                         rs1554041457   \n",
       "432377                                    -   \n",
       "432378                         rs1203405019   \n",
       "432379                         rs1436393669   \n",
       "\n",
       "                                                    Extra  \n",
       "0       IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "1       IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "2       IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "3       IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "4       IMPACT=MODERATE;STRAND=-1;CANONICAL=YES;GIVEN_...  \n",
       "...                                                   ...  \n",
       "432375  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "432376  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "432377  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "432378  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "432379  IMPACT=MODERATE;STRAND=1;CANONICAL=YES;GIVEN_R...  \n",
       "\n",
       "[432380 rows x 19 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for_search_df = df_clinvar_2.merge(df_1, on='ID', how='right')\n",
    "for_search_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "fa769b1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_num = for_search_df[(for_search_df['CLNSIG'].str.contains(\"Benign\")==True) & (for_search_df['CLNSIG'].str.contains(\"Likely_benign\")==False)].shape[0]\n",
    "likely_benign_num = for_search_df[for_search_df['CLNSIG'].str.contains(\"Likely_benign\")==True].shape[0]\n",
    "pathogenic_num = for_search_df[(for_search_df['CLNSIG'].str.contains(\"Pathogenic\")==True) & (for_search_df['CLNSIG'].str.contains(\"Likely_pathogenic\")==False)].shape[0]\n",
    "likely_pathogenic_num = for_search_df[for_search_df['CLNSIG'].str.contains(\"Likely_pathogenic\")==True].shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "79360e8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_new_num  = benign_df.shape[0]\n",
    "likely_benign_new_num  = likely_benign_df.shape[0]\n",
    "pathogenic_new_num  = pathogenic_df.shape[0]\n",
    "likely_pathogenic_new_num  = likely_pathogenic_df.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "2ea530be",
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_num_p = benign_num / df_1.shape[0] * 100\n",
    "likely_benign_num_p = likely_benign_num / df_1.shape[0] * 100\n",
    "pathogenic_num_p = pathogenic_num / df_1.shape[0] * 100 \n",
    "likely_pathogenic_num_p = likely_pathogenic_num / df_1.shape[0] * 100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "a925be03",
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_new_num_p = benign_new_num / df_6.shape[0] * 100\n",
    "likely_benign_new_num_p = likely_benign_new_num / df_6.shape[0] * 100\n",
    "pathogenic_new_num_p = pathogenic_new_num / df_6.shape[0] * 100\n",
    "likely_pathogenic_new_num_p = likely_pathogenic_new_num / df_6.shape[0] * 100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "5b6c18d4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Parent</th>\n",
       "      <th>%</th>\n",
       "      <th>New</th>\n",
       "      <th>.%</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>benign</th>\n",
       "      <td>11069</td>\n",
       "      <td>2.560017</td>\n",
       "      <td>1510</td>\n",
       "      <td>2.382342</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>likely_benign</th>\n",
       "      <td>17951</td>\n",
       "      <td>4.151672</td>\n",
       "      <td>2367</td>\n",
       "      <td>3.734440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pathogenic</th>\n",
       "      <td>16030</td>\n",
       "      <td>3.707387</td>\n",
       "      <td>2420</td>\n",
       "      <td>3.818058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>likely_pathogenic</th>\n",
       "      <td>19196</td>\n",
       "      <td>4.439613</td>\n",
       "      <td>2750</td>\n",
       "      <td>4.338703</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Parent         %   New        .%\n",
       "Index                                              \n",
       "benign              11069  2.560017  1510  2.382342\n",
       "likely_benign       17951  4.151672  2367  3.734440\n",
       "pathogenic          16030  3.707387  2420  3.818058\n",
       "likely_pathogenic   19196  4.439613  2750  4.338703"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "statistics_1_df = pd.DataFrame()\n",
    "statistics_1_df[\"Index\"] = ['benign', 'likely_benign', 'pathogenic', 'likely_pathogenic']\n",
    "statistics_1_df[\"Parent\"] = [benign_num, likely_benign_num, pathogenic_num, likely_pathogenic_num]\n",
    "statistics_1_df[\"%\"] = [benign_num_p, likely_benign_num_p, pathogenic_num_p, likely_pathogenic_num_p]\n",
    "statistics_1_df[\"New\"] = [benign_new_num, likely_benign_new_num, pathogenic_new_num, likely_pathogenic_new_num]\n",
    "statistics_1_df[\".%\"] = [benign_new_num_p, likely_benign_new_num_p, pathogenic_new_num_p, likely_pathogenic_new_num_p]\n",
    "statistics_1_df.set_index('Index', inplace=True)\n",
    "statistics_1_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8be7a301",
   "metadata": {},
   "source": [
    "Build a bar chart of the description of samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "9661a0f5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<AxesSubplot:xlabel='Index'>"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYMAAAFWCAYAAACLl/GOAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/YYfK9AAAACXBIWXMAAAsTAAALEwEAmpwYAAAlmklEQVR4nO3de5RU9Znu8e8TJIDGC2LLUUBBhxgVGEQUIuByNCLiBRwT42WEGCNRMaPRMxmMOZGoZHESjQ7Gy6AyYmJQ4mXAiGPQo6PRoKAhICChvTdDEBEFgreW9/xRv8ai7YburureXdXPZ61aXfvdu6rf+iXy9N6/XXsrIjAzs7btC1k3YGZm2XMYmJmZw8DMzBwGZmaGw8DMzHAYmJkZDQgDST0kPSFpqaQlki5O9d0lzZW0Iv3snOqSNEVSpaRFkgbkvdfYtP0KSWPz6odKWpxeM0WSmuPDmplZ3RqyZ1ANXBYRBwGDgfGSDgImAI9HRG/g8bQMcDzQOz3GAbdALjyAK4FBwOHAlTUBkrY5L+91Iwr/aGZm1lA7bG+DiFgFrErPN0haBnQDRgFHpc2mA08C/5rqd0Xu22zzJO0maa+07dyIeBdA0lxghKQngV0iYl6q3wWMBh7ZVl977LFH9OzZs+Gf1MzMeOGFF96JiIra9e2GQT5JPYFDgOeArikoAP4KdE3PuwFv5b2sKtW2Va+qo75NPXv2ZMGCBY1p38yszZP0Rl31Bk8gS/oScD9wSUSsz1+X9gKa/boWksZJWiBpwZo1a5r715mZtRkNCgNJ7ckFwd0R8UAqr06Hf0g/3071lUCPvJd3T7Vt1bvXUf+ciJgaEQMjYmBFxef2cszMrIkacjaRgDuAZRHxi7xVs4GaM4LGArPy6mPSWUWDgffT4aRHgeGSOqeJ4+HAo2ndekmD0+8ak/deZmbWAhoyZzAEOBtYLGlhqv0QmAzMlHQu8AZwWlo3BxgJVAKbgHMAIuJdSVcD89N2V9VMJgMXAncCnchNHG9z8rg+n3zyCVVVVXz44YdNeXlZ6dixI927d6d9+/ZZt2JmJUClegnrgQMHRu0J5Ndee42dd96ZLl260Ja/qhARrF27lg0bNtCrV6+s2zGzVkTSCxExsHa9rL6B/OGHH7b5IACQRJcuXbyHZGYNVlZhALT5IKjhcTCzxii7MDAzs8Zr1JfOSk3PCQ8X9f1en3zCdrdp164dffv2pbq6mgMPPJDp06ez4447FrWPGu+99x6/+c1vuPDCC5vl/c2s7SjrMMhCp06dWLhwIQBnnXUWt956K5deeul2X1ddXc0OOzTuf4733nuPm2++2WFg1kgN/UOxIX8AlgsfJmpGw4YNo7KykoceeohBgwZxyCGH8LWvfY3Vq1cDMHHiRM4++2yGDBnC2WefzZo1azj11FM57LDDOOyww3jmmWe2bPftb3+bo446iv32248pU6YAMGHCBF555RX69+/Pv/zLv2T2Oc2s9HnPoJlUV1fzyCOPMGLECIYOHcq8efOQxO23387PfvYzrrvuOgCWLl3KH/7wBzp16sSZZ57J97//fYYOHcqbb77Jcccdx7JlywB4+eWXeeKJJ9iwYQMHHHAAF1xwAZMnT+all17asidiZtZUDoMi++CDD+jfvz+Q2zM499xzWb58Od/85jdZtWoVH3/88Vbn/p988sl06tQJgMcee4ylS5duWbd+/Xo2btwIwAknnECHDh3o0KEDe+6555a9CzOzYnAYFFn+nEGN733ve1x66aWcfPLJPPnkk0ycOHHLup122mnL882bNzNv3jw6duz4ufft0KHDluft2rWjurq66L2bWdvlOYMW8P7779OtW+6q3NOnT693u+HDh3PjjTduWd7e4Z+dd96ZDRs2FKVHM2vbynrPoLWcCTBx4kS+8Y1v0LlzZ44++mhee+21OrebMmUK48ePp1+/flRXV3PkkUdy66231vu+Xbp0YciQIfTp04fjjz+en//85831EcyszJXVtYmWLVvGgQcemFFHrY/Hw6xubfnU0jZxbSIzM2sah4GZmTkMzMzMYWBmZjgMzMyMMj+11LLTmCvGluMZG2alZrthIGkacCLwdkT0SbV7gQPSJrsB70VEf0k9gWXA8rRuXkScn15zKJ/d53gOcHFEhKTdgXuBnsDrwGkRsa4Inw0m7lqUt/ns/d7f5uo1a9Zwyimn8N5773HNNdcwevRoAEaNGsUtt9zC3nvvXdx+zMyKpCGHie4ERuQXIuKbEdE/IvoD9wMP5K1+pWZdTRAktwDnAb3To+Y9JwCPR0Rv4PG0XJJmzJjB+eefz/PPP88NN9wAwEMPPcQhhxziIDCzVm27YRARTwHv1rVOuXsrngbM2NZ7SNoL2CUi5kXuW253AaPT6lFAzTUapufVS0779u3ZtGkTH3300ZbrB91www384Ac/yLo1M7NtKnQCeRiwOiJW5NV6SfqTpP+WNCzVugFVedtUpRpA14hYlZ7/FehaYE+ZOfPMM5k1axbHHnssP/zhD7n55ps5++yzm+1OZ2ZmxVLoBPIZbL1XsArYJyLWpjmC/5R0cEPfLM0h1Ht9DEnjgHEA++yzTxNbbj677rorDz+cmzhdt24dkydP5sEHH+S8885j3bp1XHbZZXz1q1/NuEszs89r8p6BpB2AfyQ3+QtARHwUEWvT8xeAV4AvAyuB7nkv755qAKvTYaSaw0lv1/c7I2JqRAyMiIEVFRVNbb1FXH311VxxxRXMmDGDoUOHMn369K0uXW1m1poUcpjoa8DLEbHl8I+kCknt0vP9yE0Uv5oOA62XNDjNM4wBZqWXzQbGpudj8+ola8WKFVRVVXHUUUexadMmvvCFLyCJDz74IOvWzMzq1JBTS2cARwF7SKoCroyIO4DT+fzE8ZHAVZI+ATYD50dEzeTzhXx2aukj6QEwGZgp6VzgDXIT0sWxnVNBm8sVV1zBpEmTADjjjDMYPXo0kydP5qqrrsqkHzOz7dluGETEGfXUv1VH7X5yp5rWtf0CoE8d9bXAMdvro5TMnDlzy/M999yTZ599NsNuzMy2z5ejMDMzh4GZmTkMzMwMh4GZmeEwMDMzHAZmZkaZ38+g7/S+RX2/xWMXb3cbSVx66aVcd911AFx77bVs3LjR3z42s1bNewZF1qFDBx544AHeeeedrFsxM2swh0GR7bDDDowbN47rr7/+c+vWrFnDqaeeymGHHcZhhx3GM888A0Dfvn157733iAi6dOnCXXfdBcCYMWOYO3dui/ZvZm2Tw6AZjB8/nrvvvpv339/6chgXX3wx3//+95k/fz73338/3/nOdwAYMmQIzzzzDEuWLGG//fbj6aefBuCPf/wjRxxxRIv3b2ZtT1nPGWRll112YcyYMUyZMoVOnTptqT/22GMsXbp0y/L69evZuHEjw4YN46mnnmLfffflggsuYOrUqaxcuZLOnTuz0047ZfERzKyNcRg0k0suuYQBAwZwzjnnbKlt3ryZefPm0bFjx622PfLII7npppt48803mTRpEg8++CD33Xcfw4YNq/221kb1nPBwg7d9ffIJzdiJlSsfJmomu+++O6eddhp33HHHltrw4cO58cYbtywvXLgQgB49evDOO++wYsUK9ttvP4YOHcq1117LkUce2dJtm1kbVdZ7Bg05FbQ5XXbZZfzyl7/csjxlyhTGjx9Pv379qK6u5sgjj+TWW28FYNCgQXz66acADBs2jMsvv5yhQ4dm0reZtT1lHQZZ2Lhx45bnXbt2ZdOmTVuW99hjD+699966XsavfvWrLc+POOIINm/e3HxNmpnV4sNEZmbmMDAzM4eBmZnRgDCQNE3S25JeyqtNlLRS0sL0GJm37nJJlZKWSzourz4i1SolTcir95L0XKrfK+mLxfyAZma2fQ3ZM7gTGFFH/fqI6J8ecwAkHQScDhycXnOzpHaS2gE3AccDBwFnpG0B/m96r78D1gHnFvKBzMys8bYbBhHxFPBuA99vFHBPRHwUEa8BlcDh6VEZEa9GxMfAPcAoSQKOBu5Lr58OjG7cRzAzs0IVcmrpRZLGAAuAyyJiHdANmJe3TVWqAbxVqz4I6AK8FxHVdWxfsGVfObBYbwXAgS8va/C2Z511FosXL+bEE0/kpz/9KQDXXHMNffr0YfTo0UXty8ysUE2dQL4F2B/oD6wCritWQ9siaZykBZIWrFmzpiV+ZZMsWrSITp06sWjRIubPn8/777/PqlWreO655xwEZtYqNSkMImJ1RHwaEZuB28gdBgJYCfTI27R7qtVXXwvsJmmHWvX6fu/UiBgYEQMrKiqa0nqLaN++PR988AGbN2/mk08+oV27dvz4xz/mJz/5SdatmZnVqUlhIGmvvMVTgJozjWYDp0vqIKkX0Bt4HpgP9E5nDn2R3CTz7IgI4Ang6+n1Y4FZTempNTnwwAOpqKhgwIABnHTSSVRWVrJ582YGDBiQdWtmZnXa7pyBpBnAUcAekqqAK4GjJPUHAngd+C5ARCyRNBNYClQD4yPi0/Q+FwGPAu2AaRGxJP2KfwXukXQN8Cfgsyu7lbAbbrhhy/OTTjqJf//3f2fSpEn8+c9/5thjj+W8887Lrjkzs1q2GwYRcUYd5Xr/wY6IScCkOupzgDl11F/ls8NMZWfWrFkceuihbNy4kVdeeYWZM2dy3HHHcdZZZ7Hjjjtm3Z6ZGeAL1TWrTz75hBtuuIGHH36YFStWkDuTFj799FM+/vhjh4GZtRplHQaNORW0WEaOHMntt9/O3nvvzU033cTYsWPZcccd6devH5s2baJv376MHDmS3XbbrcV7MzOrT1mHQRbmzPnsSNgll1yy5bkkZsyYkUFHZmbb5wvVmZmZw8DMzMowDHJfXTCPg5k1RlmFQceOHVm7dm2b/4cwIli7di0dO3bMuhUzKxFlNYHcvXt3qqqqaM3XLWopHTt2pHv37lm3YWYloqzCoH379vTq1SvrNszMSk5ZHSYyM7OmcRiYmZnDwMzMHAZmZobDwMzMcBiYmRkOAzMzw2FgZmY4DMzMjAaEgaRpkt6W9FJe7eeSXpa0SNKDknZL9Z6SPpC0MD1uzXvNoZIWS6qUNEXptl+Sdpc0V9KK9LNzM3xOMzPbhobsGdwJjKhVmwv0iYh+wF+Ay/PWvRIR/dPj/Lz6LcB5QO/0qHnPCcDjEdEbeDwtm5lZC9puGETEU8C7tWq/j4jqtDgP2OYV0STtBewSEfMid0nRu4DRafUoYHp6Pj2vbmZmLaQYcwbfBh7JW+4l6U+S/lvSsFTrBlTlbVOVagBdI2JVev5XoGsRejIzs0Yo6Kqlkq4AqoG7U2kVsE9ErJV0KPCfkg5u6PtFREiq92YEksYB4wD22WefpjduZmZbafKegaRvAScCZ6VDP0TERxGxNj1/AXgF+DKwkq0PJXVPNYDV6TBSzeGkt+v7nRExNSIGRsTAioqKprZuZma1NCkMJI0AfgCcHBGb8uoVktql5/uRmyh+NR0GWi9pcDqLaAwwK71sNjA2PR+bVzczsxay3cNEkmYARwF7SKoCriR39lAHYG46Q3ReOnPoSOAqSZ8Am4HzI6Jm8vlCcmcmdSI3x1AzzzAZmCnpXOAN4LSifDIzM2uw7YZBRJxRR/mOera9H7i/nnULgD511NcCx2yvDzMzaz7+BrKZmTkMzMzMYWBmZjgMzMwMh4GZmeEwMDMzHAZmZobDwMzMcBiYmRkOAzMzw2FgZmYUeD+DctNzwsMN2u71ySc0cydmZi3LewZmZuYwMDMzh4GZmeEwMDMzHAZmZobDwMzMaGAYSJom6W1JL+XVdpc0V9KK9LNzqkvSFEmVkhZJGpD3mrFp+xWSxubVD5W0OL1mitKNlc3MrGU0dM/gTmBErdoE4PGI6A08npYBjgd6p8c44BbIhQdwJTAIOBy4siZA0jbn5b2u9u8yM7Nm1KAwiIingHdrlUcB09Pz6cDovPpdkTMP2E3SXsBxwNyIeDci1gFzgRFp3S4RMS8iArgr773MzKwFFDJn0DUiVqXnfwW6pufdgLfytqtKtW3Vq+qom5lZCynKBHL6iz6K8V7bImmcpAWSFqxZs6a5f52ZWZtRSBisTod4SD/fTvWVQI+87bqn2rbq3euof05ETI2IgRExsKKiooDWzcwsXyFhMBuoOSNoLDArrz4mnVU0GHg/HU56FBguqXOaOB4OPJrWrZc0OJ1FNCbvvczMrAU06KqlkmYARwF7SKoid1bQZGCmpHOBN4DT0uZzgJFAJbAJOAcgIt6VdDUwP213VUTUTEpfSO6MpU7AI+lhZmYtpEFhEBFn1LPqmDq2DWB8Pe8zDZhWR30B0KchvZiZWfH5G8hmZuYwMDMzh4GZmeEwMDMzHAZmZobDwMzMcBiYmRkOAzMzw2FgZmY4DMzMDIeBmZnhMDAzMxwGZmaGw8DMzHAYmJkZDgMzM8NhYGZmOAzMzIwCwkDSAZIW5j3WS7pE0kRJK/PqI/Nec7mkSknLJR2XVx+RapWSJhT6oczMrHEadA/kukTEcqA/gKR2wErgQeAc4PqIuDZ/e0kHAacDBwN7A49J+nJafRNwLFAFzJc0OyKWNrU3MzNrnCaHQS3HAK9ExBuS6ttmFHBPRHwEvCapEjg8rauMiFcBJN2TtnUYmJm1kGLNGZwOzMhbvkjSIknTJHVOtW7AW3nbVKVafXUzM2shBYeBpC8CJwO/TaVbgP3JHUJaBVxX6O/I+13jJC2QtGDNmjXFelszszavGHsGxwMvRsRqgIhYHRGfRsRm4DY+OxS0EuiR97ruqVZf/XMiYmpEDIyIgRUVFUVo3czMoDhhcAZ5h4gk7ZW37hTgpfR8NnC6pA6SegG9geeB+UBvSb3SXsbpaVszM2shBU0gS9qJ3FlA380r/0xSfyCA12vWRcQSSTPJTQxXA+Mj4tP0PhcBjwLtgGkRsaSQvszMrHEKCoOI+BvQpVbt7G1sPwmYVEd9DjCnkF7MzKzp/A1kMzNzGJiZmcPAzMxwGJiZGQ4DMzPDYWBmZjgMzMwMh4GZmeEwMDMzHAZmZobDwMzMcBiYmRkOAzMzw2FgZmY4DMzMDIeBmZnhMDAzMxwGZmZGEcJA0uuSFktaKGlBqu0uaa6kFeln51SXpCmSKiUtkjQg733Gpu1XSBpbaF9mZtZwxdoz+IeI6B8RA9PyBODxiOgNPJ6WAY4HeqfHOOAWyIUHcCUwCDgcuLImQMzMrPk112GiUcD09Hw6MDqvflfkzAN2k7QXcBwwNyLejYh1wFxgRDP1ZmZmtRQjDAL4vaQXJI1Lta4RsSo9/yvQNT3vBryV99qqVKuvbmZmLWCHIrzH0IhYKWlPYK6kl/NXRkRIiiL8HlLYjAPYZ599ivGWZmZGEfYMImJl+vk28CC5Y/6r0+Ef0s+30+YrgR55L++eavXVa/+uqRExMCIGVlRUFNq6mZklBYWBpJ0k7VzzHBgOvATMBmrOCBoLzErPZwNj0llFg4H30+GkR4HhkjqniePhqWZmZi2g0MNEXYEHJdW8128i4r8kzQdmSjoXeAM4LW0/BxgJVAKbgHMAIuJdSVcD89N2V0XEuwX2ZmZmDVRQGETEq8Df11FfCxxTRz2A8fW81zRgWiH9mJlZ0/gbyGZm5jAwMzOHgZmZ4TAwMzMcBmZmhsPAzMxwGJiZGQ4DMzPDYWBmZjgMzMwMh4GZmeEwMDMzHAZmZobDwMzMcBiYmRkOAzMzw2FgZmY4DMzMjALCQFIPSU9IWippiaSLU32ipJWSFqbHyLzXXC6pUtJyScfl1UekWqWkCYV9JDMza6xC7oFcDVwWES9K2hl4QdLctO76iLg2f2NJBwGnAwcDewOPSfpyWn0TcCxQBcyXNDsilhbQm5lZi+o7vW+Dtls8dnEzd9I0TQ6DiFgFrErPN0haBnTbxktGAfdExEfAa5IqgcPTusqIeBVA0j1pW4eBmVkLKcqcgaSewCHAc6l0kaRFkqZJ6pxq3YC38l5WlWr11c3MrIUUHAaSvgTcD1wSEeuBW4D9gf7k9hyuK/R35P2ucZIWSFqwZs2aYr2tmVmbV8icAZLakwuCuyPiAYCIWJ23/jbgd2lxJdAj7+XdU41t1LcSEVOBqQADBw6MQnq30tPQY7LQeo/LtiYeT8tXyNlEAu4AlkXEL/Lqe+VtdgrwUno+GzhdUgdJvYDewPPAfKC3pF6Svkhuknl2U/syM7PGK2TPYAhwNrBY0sJU+yFwhqT+QACvA98FiIglkmaSmxiuBsZHxKcAki4CHgXaAdMiYkkBfZmZWSMVcjbRHwDVsWrONl4zCZhUR33Otl5nZmbNy99ANjMzh4GZmTkMzMwMh4GZmeEwMDMzHAZmZkaB30C27Sv1KxmaWdvgPQMzM3MYmJmZw8DMzHAYmJkZDgMzM8NhYGZmOAzMzAyHgZmZ4TAwMzMcBmZmhsPAzMxoRWEgaYSk5ZIqJU3Iuh8zs7akVYSBpHbATcDxwEHAGZIOyrYrM7O2o1WEAXA4UBkRr0bEx8A9wKiMezIzazNaSxh0A97KW65KNTMzawGKiKx7QNLXgRER8Z20fDYwKCIuqrXdOGBcWjwAWN6ijTbNHsA7WTdRJjyWxeXxLK5SGc99I6KidrG13NxmJdAjb7l7qm0lIqYCU1uqqWKQtCAiBmbdRznwWBaXx7O4Sn08W8thovlAb0m9JH0ROB2YnXFPZmZtRqvYM4iIakkXAY8C7YBpEbEk47bMzNqMVhEGABExB5iTdR/NoKQOa7VyHsvi8ngWV0mPZ6uYQDYzs2y1ljkDMzPLkMPAzMwcBmZm1oomkMuJpA7AqUBP8sY4Iq7KqqdS5vEsHkm9gFUR8WFa7gR0jYjXM22sRJXTeHrPoHnMIndtpWrgb3kPaxqPZ/H8Ftict/xpqlnTlM14es+geXSPiBFZN1FGPJ7Fs0O6GCQAEfFx+qKnNU3ZjKf3DJrHs5L6Zt1EGfF4Fs8aSSfXLEgaRWlcT6e1Kpvx9PcMmoGkpcDfAa8BHwECIiL6ZdpYifJ4Fo+k/YG7gb3JjeNbwJiIqMy0sRJVTuPpMGgGkvatqx4Rb7R0L+XA41l8kr4EEBEbs+6lHJTDeHrOoHlsaGDNGsbjWSBJ/xQRv5Z0aa06ABHxi0waK1HlOJ4Og+bxIrlLcq8jt+u4G/BXSauB8yLihQx7K0Uez8LtlH7unGkX5aPsxtOHiZqBpNuA+yLi0bQ8nNx58v8B/FtEDMqyv1Lj8TRrfg6DZiBpcUT0rVVbFBH9JC2MiP4ZtVaSPJ7FI6kCOI/Pf4Hv21n1VMrKaTx9mKh5rJL0r8A9afmbwGpJ7dj6CyrWMB7P4pkFPA08Ru4LUlaYshlP7xk0A0l7AFcCQ1PpGeAnwPvAPqV42lmWPJ7F4z2p4iqn8XQYmLUhkq4Bnk03k7ICldN4OgyKSNINEXGJpIeAzw1sRJxcx8usHh7P4pO0gdyZMB+nR80X+HbJtLESVU7j6TmD4vpV+nltpl2UD49nkUVE2ZwK2RqU03h6z8CsDVHuW1FnAb0i4mpJPYC9IuL5jFsrSeU0ng6DZiBpCDAR2Jfc3lfNruN+WfZVqjyexSPpFnJnYB0dEQdK6gz8PiIOy7i1klRO4+nDRM3jDuD7wAuU+OlmrYTHs3gGRcQASX8CiIh1pXrJ5VaibMbTYdA83o+IR7Juoox4PIvnk/T9jIAtX5rydzWarmzG02HQPJ6Q9HPgAXKXXAYgIl7MrqWS5vEsninAg8CekiYBXwd+lG1LJa1sxtNzBs1A0hN1lCMijm7xZsqAx7O4JH0FOIbc3MvjEbEs45ZKWrmMp8PArA2RtHsd5Q0R8UmLN1MGymk8fdvLZiCpq6Q7JD2Slg+SdG7WfZUqj2dRvQisAf4CrEjPX5f0oqRDM+2sNJXNeDoMmsedwKPkboUHuf+jXJJVM2XgTjyexTIXGBkRe0REF+B44HfAhcDNmXZWmspmPB0GzWOPiJhJOqsgIqrxKZGF8HgWz+Ca+0IARMTvga9GxDygQ3ZtlayyGU+fTdQ8/iapC5+dbjaY3BU2rWk8nsXjy4EXV9mMpyeQm4GkAcCNwMHAEqAC+HpELMq0sRKVN559gJfweDaZLwdeXOU0ng6DZiCpI3ARcBy5G7f/EbgxIj7MtLESJmkH4AByp+8tL8WzNVoTSTuTOz13Y9a9lINyGE+HQTOQNBNYD9ydSmcCu0XEN7LrqnSlcL2Q3F9fQe7OUrc6XBtPUl/gLqDmlMh3gLER8VJ2XZWuchpPh0EzkLQ0Ig7aXs0aJoXrBuDXqeRwbSJJzwJXRMQTafko4KcRcUSWfZWqchpPTyA3jxclDU5nFCBpELAg455KWZ9aQfqEpKWZdVPadqr5hwsgIp6UtFOWDZW4shlPh0ERSVpM7jBGe+BZSW+m5X2Bl7PsrcQ5XIvnVUn/h89uHPRPwKsZ9lPqymY8fZioiCTtu631EfFGS/VSDmqF6wHAVuHqw26Nl663/xM+O/vlaWBiRKzLrqvSVU7j6TCwVquh4Sqpcyn+x2fWmjgMrORJejEiBmTdRymQ9GXgfwM9yTtM7CvANk05jafnDKwcKOsGSshvgVuB2/ElPYqhbMbTYWDlwLu3DVcdEbdk3UQZKZvxdBiYtQF5191/SNKF5O7OlX/XuHczaaxEleN4es7ASp6kP0XEIVn30ZpJeo3cHlRdh9QiIvZr4ZZKWjmOp8PAWj1J1wHTImJJPet3L8W/xLIgqWPty3jUVbOGKafx9P0MrBQsA6ZKek7S+ZJ2zV/pIGiUZxtYs4Ypm/H0nIG1ehFxO3C7pAOAc4BFkp4Bbsu/FIDVT9L/AroBnSQdwmeHN3YBdsyssRJVjuPpMLCSkG4W8pX0eAf4M3CppO9GxOmZNlcajgO+BXQHfpFX3wD8MIuGSlzZjafnDKzVk3Q9cBLwOHBHRDyft255RByQWXMlRtKpEXF/1n2Ui3IaT4eBtXqSzgFmRsTf6li3a0T4FpiNIOkEcnfh61hTi4irsuuotJXLePowkbVa6XaXkDskdIC09Vl8EfGig6BxJN1K7pj2P5D71uzXgee3+SKrVzmNp/cMrNWStK3J4SjF679kTdKiiOiX9/NLwCMRMSzr3kpROY2n9wys1YqIf8i6hzL0Qfq5SdLewFpgrwz7KXVlM57+noG1epJ2lPQjSVPTcm9JJ2bdV4n6naTdgJ8BLwCvAzOybKjE1Yznz4EXyY3nb7JsqKl8mMhaPUn3kvuHa0xE9JG0I/BsRPTPtrPSI6kTcAEwjNzlFJ4GbinFb8y2NpI6AB1LdR7LYWCtnqQFETEw/xpEkv4cEX+fdW+lRtJMcufC/zqVzgR2jYjTsuuqdEnqCFxI7k5nAfyBEg1XzxlYKfg4/UUbAJL2J+8KkdYofWrdLvQJSUsz66b03UUuXG9My2eSux/yNzLrqIkcBlYKrgT+C+gh6W5gCLlvf1rjvShpcETMA5A0CFiQcU+lrGzC1WFgpeAF4B+BweSuAXMxsHOmHZWuQ4FnJb2ZlvcBlktaTO503X7ZtVaSyiZcPWdgrV66KN3xEbE+LR8I/DYi+mTbWemRtO+21kfEGy3VSzmQtAw4ANgqXIFqSixcHQbW6qWv+/8AGEnuQnV3AWdFxMIs+zLbXrgC6yNiXYs0UyAfJrJWLyIeltQemEvu8NApEfGXjNsy2+6elKQXgQHb2qa1cBhYqyXpRra+2f2uwCvARZKIiH/OpjOzBqvrtpitksPAWrPaE3EvZNKFWdOVzHF4h4G1WhExPesezNoKh4G1WpJmRsRpNac91l5fSmdqWJtVMoeJfDaRtVqS9oqIVfWdseHTIC1rkq4DpkXEknrW7x4R77ZwW03iMDAzayJJ3wHOIXeU5T+AGb5QnVmRSdpA3RNwIveFnl1auCWzOkk6gFwonAE8A9wWEdu6OVOr4zAwMyuApHbAieTCoAcwk9xVTP8WEadn2VtjOAzMzJpI0vXAScDjwB0R8XzeuuURcUBmzTWSzyYyM2u6RcCPIuJvdaw7vKWbKYT3DMzMGknSNi8xEREvtlQvxeIwMDNrJEnbmhyOiDi6xZopEoeBmZnxhawbMDMrVZJ2lPQjSVPTcm9JJ2bdV1M4DMzMmu4/gI+BI9LySuCa7NppOoeBmVnT7R8RPwM+AYiITZTQ9YjyOQzMzJruY0mdSN+Ul7Q/8FG2LTWNv2dgZtZ0VwL/BfSQdDcwBPhWph01kc8mMjNrIkm7kzssNDj9nAfsHBGvZdpYEzgMzMyaSNIzwPERsT4tHwj8NiL6ZNtZ43nOwMys6X4KPCRpJ0mHAvcB/5RxT03iOQMzsyaKiIcltQfmAjsDp0TEXzJuq0l8mMjMrJEk3cjW99o4BngFeB0gIv45g7YK4j0DM7PGW1Br+YVMuigi7xmYmZn3DMzMGkvSzIg4TdJi6rg1a0T0y6CtgnjPwMyskSTtFRGrJO1b1/qIeKOleyqUw8DMzHyYyMyssSRtoI7DQ+S+hRwRsUsLt1Qw7xmYmZm/gWxmZg4DMzPDYWC2TZI2NnL7oyT9rrn6MWsuDgMzM3MYmDVE+ov/SUn3SXpZ0t2SlNaNSLUXgX/Me81OkqZJel7SnySNSvV/k/Tj9Pw4SU9J8n+LlimfWmrWcIcABwP/AzwDDJG0ALgNOBqoBO7N2/4K4P9FxLcl7QY8L+kx4HJgvqSngSnAyIjY3HIfw+zz/NeIWcM9HxFV6R/uhUBP4CvAaxGxInLnaf86b/vhwARJC4EngY7APumm6eeRu+zxLyPilRb7BGb18J6BWcPl3+j8U7b/34+AUyNieR3r+gJrgb2L1JtZQbxnYFaYl4GekvZPy2fkrXsU+F7e3MIh6ee+wGXkDjsdL2lQC/ZrVieHgVkBIuJDYBzwcJpAfjtv9dVAe2CRpCXA1SkY7gD+d0T8D3AucLukji3cutlWfDkKMzPznoGZmTkMzMwMh4GZmeEwMDMzHAZmZobDwMzMcBiYmRkOAzMzA/4/BpxVVoIyYicAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "statistics_1_df.plot.bar()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "0ddfcac1",
   "metadata": {},
   "outputs": [],
   "source": [
    "a = len(omim_list) # List of OMIM genes\n",
    "b = search_omim_df.shape[0] # OMIM contained in clinvar_vep_ref.vcf\n",
    "c = omim_missence_canonical_df.shape[0] # OMIM missence and canonical variants\n",
    "d = df_6.shape[0] # OMIM after script\n",
    "e = df_1.shape[0] # OMIM before script\n",
    "f = df_5.shape[0] # before splitting"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7699020",
   "metadata": {},
   "source": [
    "OMIM missence and canonical variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "9d441fc5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>number</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>OMIM list</th>\n",
       "      <td>3950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OMIM clinvar ann</th>\n",
       "      <td>1503664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>filtred</th>\n",
       "      <td>432380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>after script</th>\n",
       "      <td>63383</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   number\n",
       "index                    \n",
       "OMIM list            3950\n",
       "OMIM clinvar ann  1503664\n",
       "filtred            432380\n",
       "after script        63383"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "statistics_2_df = pd.DataFrame()\n",
    "statistics_2_df[\"index\"] = ['OMIM list', 'OMIM clinvar ann', 'filtred', 'after script']\n",
    "statistics_2_df[\"number\"] = [a, b, c, d]\n",
    "statistics_2_df.set_index('index', inplace=True)\n",
    "statistics_2_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "000ff6e1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<AxesSubplot:xlabel='index'>"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAFeCAYAAACVVOQDAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/YYfK9AAAACXBIWXMAAAsTAAALEwEAmpwYAAAdnUlEQVR4nO3de5hddX3v8feHi8RLBIX0aUvAUBspIEg0IopVQFGQI7RqOaT2eIGSY6vWVrGltWgPHHus9dgLgpYWRGjLReBIjqLYIoo3LkGQclNTQBn0HEIEtQJC4Ns/9pqwGSYzk8zes7JX3q/nyTOz1vrN7O/sJ/lkzW/9LqkqJEmjb4u2C5AkDYaBLkkdYaBLUkcY6JLUEQa6JHWEgS5JHdFqoCc5PcldSW6YYfsjktyU5MYk/zzs+iRplKTNcehJXgL8B3BmVT17mraLgfOAA6vqniQ/V1V3zUWdkjQKWr1Dr6rLgR/2n0vyzCSfS3JNki8n+ZXm0jHAyVV1T/O1hrkk9dkU+9BPBd5eVc8DjgVOac4/C3hWkq8muSLJwa1VKEmboK3aLqBfkqcALwI+mWT89DbNx62AxcD+wELg8iR7VtW9c1ymJG2SNqlAp/cbw71Vtfck18aAK6vqIeC2JN+mF/BXz2F9krTJ2qS6XKrqx/TC+jcA0vOc5vKn6N2dk2QHel0wt7ZQpiRtktoetng28HVg1yRjSY4GXg8cneSbwI3A4U3zS4A1SW4CLgPeXVVr2qhbkjZFrQ5blCQNzibV5SJJ2ngGuiR1RGujXHbYYYdatGhRWy8vSSPpmmuuubuqFkx2rbVAX7RoEStXrmzr5SVpJCX57vqu2eUiSR1hoEtSRxjoktQRm9rUf0kd99BDDzE2NsYDDzzQdimbtHnz5rFw4UK23nrrGX+NgS5pTo2NjTF//nwWLVpE3yJ86lNVrFmzhrGxMXbZZZcZf51dLpLm1AMPPMD2229vmE8hCdtvv/0G/xZjoEuac4b59DbmPTLQJWlA9t9//1bn19iHrvVadNxn2i5hRm7/wKFtl6BZGPTfs1H9+7B27Vq22mp2kewduqTNzu23385uu+3GMcccwx577MErXvEK7r///sfcYd99992ML09yxhln8Gu/9mscdNBBLFq0iI985CN8+MMfZsmSJey777788IePbo181llnsffee/PsZz+bq666CoCf/vSnHHXUUeyzzz4sWbKEiy66aN33PeywwzjwwAN52cteNuufy0CXtFn6zne+w1vf+lZuvPFGtttuOy644IIp299www1ceOGFXH311bznPe/hSU96Etdeey0vfOELOfPMM9e1u++++7juuus45ZRTOOqoowB4//vfz4EHHshVV13FZZddxrvf/W5++tOfAvCNb3yD888/ny996Uuz/pnscpG0Wdpll13Ye++9AXje857H7bffPmX7Aw44gPnz5zN//ny23XZbXv3qVwOw5557cv31169rt2zZMgBe8pKX8OMf/5h7772Xz3/+86xYsYIPfehDQG+kz/e+9z0ADjroIJ7+9KcP5Gcy0CVtlrbZZpt1n2+55Zbcf//9bLXVVjzyyCMAjxsy2N9+iy22WHe8xRZbsHbt2nXXJo5OSUJVccEFF7Drrrs+5tqVV17Jk5/85MH8QNjlIknrLFq0iGuuuQaA888/f6O+x7nnngvAV77yFbbddlu23XZbXvnKV3LSSScxvkPctddeO5iCJzDQJalx7LHH8tGPfpQlS5Zw9913b9T3mDdvHkuWLOEtb3kLp512GgDHH388Dz30EHvttRd77LEHxx9//CDLXqe1PUWXLl1aroe+aXPYoobh5ptvZrfddmu7jJEw2XuV5JqqWjpZe+/QJakjDHRJ6ggDXZI6YtpAT3J6kruS3DBNu+cnWZvkdYMrT1IXtfXsbpRszHs0kzv0M4CDp2qQZEvgL4DPb3AFkjYr8+bNY82aNYb6FMbXQ583b94Gfd20E4uq6vIki6Zp9nbgAuD5G/TqkjY7CxcuZGxsjNWrV7ddyiZtfMeiDTHrmaJJdgR+HTiAaQI9yXJgOcDOO+8825eWNIK23nrrDdqFRzM3iIeifw38UVU9Ml3Dqjq1qpZW1dIFCxYM4KUlSeMGsZbLUuCcZv2CHYBXJVlbVZ8awPeWJM3QrAO9qtb97pTkDODThrkkzb1pAz3J2cD+wA5JxoD3AVsDVNXHhlqdJGnGZjLKZdlMv1lVvWlW1UiSNpozRSWpIwx0SeoIA12SOsJAl6SOMNAlqSMMdEnqCANdkjrCQJekjjDQJakjDHRJ6ggDXZI6wkCXpI4w0CWpIwx0SeoIA12SOsJAl6SOMNAlqSMMdEnqiGkDPcnpSe5KcsN6rr8+yfVJ/i3J15I8Z/BlSpKmM5M79DOAg6e4fhvw0qraEzgROHUAdUmSNtBMNom+PMmiKa5/re/wCmDhAOqSJG2gQfehHw18dsDfU5I0A9Peoc9UkgPoBfqLp2izHFgOsPPOOw/qpSVJDOgOPclewD8Ah1fVmvW1q6pTq2ppVS1dsGDBIF5aktSYdaAn2Rm4EPhvVfXt2ZckSdoY03a5JDkb2B/YIckY8D5ga4Cq+hjwXmB74JQkAGuraumwCpYkTW4mo1yWTXP9t4HfHlhFkqSN4kxRSeoIA12SOsJAl6SOMNAlqSMMdEnqCANdkjrCQJekjjDQJakjDHRJ6ggDXZI6wkCXpI4w0CWpIwx0SeoIA12SOsJAl6SOMNAlqSMMdEnqCANdkjrCQJekjpg20JOcnuSuJDes53qS/G2SVUmuT/LcwZcpSZrOTO7QzwAOnuL6IcDi5s9y4KOzL0uStKGmDfSquhz44RRNDgfOrJ4rgO2S/MKgCpQkzcwg+tB3BO7oOx5rzj1OkuVJViZZuXr16gG8tCRp3Jw+FK2qU6tqaVUtXbBgwVy+tCR13iAC/U5gp77jhc05SdIcGkSgrwDe0Ix22Rf4UVX9YADfV5K0AbaarkGSs4H9gR2SjAHvA7YGqKqPARcDrwJWAfcBbx5WsZKk9Zs20Ktq2TTXC3jrwCqSJG0UZ4pKUkcY6JLUEQa6JHWEgS5JHWGgS1JHGOiS1BEGuiR1hIEuSR1hoEtSRxjoktQRBrokdYSBLkkdYaBLUkcY6JLUEQa6JHWEgS5JHWGgS1JHGOiS1BEzCvQkByf5VpJVSY6b5PrOSS5Lcm2S65O8avClSpKmMm2gJ9kSOBk4BNgdWJZk9wnN/hQ4r6qWAEcCpwy6UEnS1GZyh74PsKqqbq2qB4FzgMMntCngqc3n2wLfH1yJkqSZ2GoGbXYE7ug7HgNeMKHNnwGfT/J24MnAywdSnSRpxgb1UHQZcEZVLQReBZyV5HHfO8nyJCuTrFy9evWAXlqSBDML9DuBnfqOFzbn+h0NnAdQVV8H5gE7TPxGVXVqVS2tqqULFizYuIolSZOaSaBfDSxOskuSJ9B76LliQpvvAS8DSLIbvUD3FlyS5tC0gV5Va4G3AZcAN9MbzXJjkhOSHNY0exdwTJJvAmcDb6qqGlbRkqTHm8lDUarqYuDiCefe2/f5TcB+gy1NkrQhnCkqSR1hoEtSRxjoktQRBrokdYSBLkkdYaBLUkcY6JLUEQa6JHWEgS5JHWGgS1JHGOiS1BEGuiR1hIEuSR1hoEtSRxjoktQRBrokdYSBLkkdYaBLUkcY6JLUETMK9CQHJ/lWklVJjltPmyOS3JTkxiT/PNgyJUnTmXaT6CRbAicDBwFjwNVJVjQbQ4+3WQz8MbBfVd2T5OeGVbAkaXIzuUPfB1hVVbdW1YPAOcDhE9ocA5xcVfcAVNVdgy1TkjSdmQT6jsAdfcdjzbl+zwKeleSrSa5IcvBk3yjJ8iQrk6xcvXr1xlUsSZrUoB6KbgUsBvYHlgF/n2S7iY2q6tSqWlpVSxcsWDCgl5YkwcwC/U5gp77jhc25fmPAiqp6qKpuA75NL+AlSXNkJoF+NbA4yS5JngAcCayY0OZT9O7OSbIDvS6YWwdXpiRpOtMGelWtBd4GXALcDJxXVTcmOSHJYU2zS4A1SW4CLgPeXVVrhlW0JOnxph22CFBVFwMXTzj33r7PC3hn80eS1AJnikpSRxjoktQRBrokdYSBLkkdYaBLUkcY6JLUEQa6JHWEgS5JHWGgS1JHGOiS1BEGuiR1hIEuSR1hoEtSRxjoktQRBrokdYSBLkkdYaBLUkcY6JLUETPagk7S7C067jNtlzCt2z9waNslaBZmdIee5OAk30qyKslxU7R7bZJKsnRwJUqSZmLaQE+yJXAycAiwO7Asye6TtJsPvAO4ctBFSpKmN5M79H2AVVV1a1U9CJwDHD5JuxOBvwAeGGB9kqQZmkmg7wjc0Xc81pxbJ8lzgZ2qaspOwiTLk6xMsnL16tUbXKwkaf1mPcolyRbAh4F3Tde2qk6tqqVVtXTBggWzfWlJUp+ZBPqdwE59xwubc+PmA88GvpjkdmBfYIUPRiVpbs0k0K8GFifZJckTgCOBFeMXq+pHVbVDVS2qqkXAFcBhVbVyKBVLkiY1baBX1VrgbcAlwM3AeVV1Y5ITkhw27AIlSTMzo4lFVXUxcPGEc+9dT9v9Z1+WJGlDOfVfkjrCQJekjjDQJakjDHRJ6ggDXZI6wkCXpI4w0CWpIwx0SeoIA12SOsJAl6SOMNAlqSMMdEnqCANdkjrCQJekjjDQJakjDHRJ6ggDXZI6wkCXpI6YUaAnOTjJt5KsSnLcJNffmeSmJNcnuTTJMwZfqiRpKtMGepItgZOBQ4DdgWVJdp/Q7FpgaVXtBZwPfHDQhUqSpjaTO/R9gFVVdWtVPQicAxze36CqLquq+5rDK4CFgy1TkjSdmQT6jsAdfcdjzbn1ORr47GyKkiRtuK0G+c2S/BawFHjpeq4vB5YD7LzzzoN8aUna7M3kDv1OYKe+44XNucdI8nLgPcBhVfWzyb5RVZ1aVUuraumCBQs2pl5J0nrMJNCvBhYn2SXJE4AjgRX9DZIsAf6OXpjfNfgyJUnTmTbQq2ot8DbgEuBm4LyqujHJCUkOa5r9JfAU4JNJrkuyYj3fTpI0JDPqQ6+qi4GLJ5x7b9/nLx9wXZKkDeRMUUnqCANdkjrCQJekjjDQJakjDHRJ6ggDXZI6wkCXpI4w0CWpIwx0SeoIA12SOsJAl6SOMNAlqSMMdEnqiIHuWCRJc2HRcZ9pu4QZuf0Dh87p63mHLkkdYaBLUkcY6JLUEQa6JHWEgS5JHTGjQE9ycJJvJVmV5LhJrm+T5Nzm+pVJFg28UknSlKYN9CRbAicDhwC7A8uS7D6h2dHAPVX1y8BfAX8x6EIlSVObyR36PsCqqrq1qh4EzgEOn9DmcOATzefnAy9LksGVKUmazkwmFu0I3NF3PAa8YH1tqmptkh8B2wN39zdKshxY3hz+R5JvbUzRc2wHJvwcmpWBv5/ZfH8f9L0crFF5P5+xvgtzOlO0qk4FTp3L15ytJCuramnbdXSF7+fg+F4OVhfez5l0udwJ7NR3vLA5N2mbJFsB2wJrBlGgJGlmZhLoVwOLk+yS5AnAkcCKCW1WAG9sPn8d8IWqqsGVKUmazrRdLk2f+NuAS4AtgdOr6sYkJwArq2oFcBpwVpJVwA/phX5XjFQX0Qjw/Rwc38vBGvn3M95IS1I3OFNUkjrCQJekjjDQJW2WkrxjJudGiYE+QfL4qQCTndPMJNkvyb8k+XaSW5PcluTWtuuSeHRkXr83zXURg+RD0QmSfKOqnjvh3PVVtVdbNY2yJLcAfwBcAzw8fr6qnKewAZL8X2C9/1ir6rA5LGekJVkG/CbwYuDLfZfmA49U1ctaKWwA3FO0keR3gN8FfinJ9X2X5gNfbaeqTvhRVX227SI64EPNx9cAPw/8Y3O8DPj/rVQ0ur4G/IDeVP//3Xf+J8D1k37FiPAOvZFkW+BpwP8C+pcI/klV/bCdqkZfkg/Qm79wIfCz8fNV9Y3Wihphk01P78KU9bYk+Xl6CxAWcHVV/b+WS5oVA32CJM8ExqrqZ0n2B/YCzqyqe9usa1QluWyS01VVB855MR2Q5Gbg0Kq6tTneBbi4qnZrt7LRk+Ro4H3AF4AALwVOqKrTWy1sFgz0CZJcBywFFgEXAxcBe1TVq1osSwJ6m83Qm9F4K70Qegbw36vqklYLG0HNaq8vGn+ek2R74GtVtWu7lW08+9Af75FmuYPXACdV1UlJrm27qFGVZBvgtfT+g1z3962qTmirplFWVZ9Lshj4lebULVX1s6m+Ruu1hl6/+bifMOKLChroj/dQ8xT8DcCrm3Nbt1jPqLsI+BG9US4GzywleRLwTuAZVXVMksVJdq2qT7dd2whaBVyZ5CJ6feiHA9cneSdAVX24zeI2hoH+eG8G3gK8v6pua/ooz2q5plG2sKoObruIDvk4vf8cX9gc3wl8EjDQN9y/N3/GXdR8nN9CLQNhH7qGKsmp9Lqu/q3tWrpgfERLkmuraklz7ptV9Zy2a1P7vENvJDmvqo5I8m9MMoHDiUUb7cXAm5LcRq/LJfRGufh+bpwHkzyR5u9oMyrLrqwNkOSvq+r31zdZa5QnaRnojxpfw+G/tFpF9xzSdgEd8z7gc8BOSf4J2I8Rn67egvEu1A9N2WoE2eWiOZHk54B548dV9b0WyxlJSbagtyPYpcC+9H7buaKq3MR8AyXZkt78kte3XcsgGeiNJD9h8rUyxrsInjrHJXVCksPoTa/+ReAueuOmb66qPVotbEQ5K3RwknwFOLCqHmy7lkGxy6VRVSP7ZHsTdyK9u8l/raolSQ4AfqvlmkbZvyY5FjgX+On4SZen2Ci3Al9NsoLHvpcjN1xxnIGuYXuoqtYk2SLJFlV1WZK/bruoEfZfm49v7TtXwC+1UMuoGx+2uAUjPFSxn4GuYbs3yVOAy4F/SnIXfXdD2mC7VdUD/SeSzFtfY61fVf2PtmsYNDe40LAdDtxHb030z9G7I3r1lF+hqXxthuc0jWbjle36jp+WZKTXxPEOXUNVVeN3448An2izllHWLPO6I/DEJEvoPawHeCrwpNYKG20L+ldRrap7mtFYI8tAb0wyyiXNsaNctCl4Jb3x5guB/od2PwH+pI2COuDhJDuPD6FN8gym2BVqFDhssZHkU/R2grkQOMdx0toUJXltVV3Qdh1d0LcU8Zfo3bj9KrB8lJciNtD7NLsWvQY4kt4kmHPphbtDwjZCVydvtGF8BcD1GeWhdm1KsgO9YbXQgUlaPhTtU1U/qqqP05uu/nfACTiteqNV1cPAM5I8oe1aOmD+NH+0gZLsB9zfLD28HfAnTbfLyPIOvU+SF9HbdPdXga8A51bVl6f+Kk0lyZnAbkBnJm+oG5rN4J9Db5vJjwOnAUdU1UtbLWwWfCjaSHI7cC9wDrAcWNucfy64qfEsdG7yRhuS/GFVfTDJSUy+QuDvtVDWqFtbVZXkcODkqjqt2Wd0ZHmH3kjyRdb/hNtNjdWqJGuqavskvw/cM/F6VTkkdAMl+RK9uRFvBl5Cb62hb1bVnq0WNgsGuoYqyQLgD4E9eOxqi/4HuQGS3AS8HPgssD+PjkMHXMtlYzRj+38TuLqqvpxkZ2D/qjqz5dI2moHeaDaFXq+qunCuaumSJJ+nN1roWHpb+70RWF1Vf9RqYSMmyduB36W3Zsud/Zfo/QbpWi4y0Mcl+fgUl6uqjpqzYjokyTVV9bwk14/vUpTk6qp6ftu1jaIkH62q32m7Dm2afCjaqKo3t11DRz3UfPxBkkOB7wNPb7GekWaYayoGesOJG0PzP5sJW+8CTqK39sgftFuSNnddnfRml0sjySPAdfQeOo1vZrxOF5fanAtJFlTV6rbrkCZyx6JuW0JvUtGhwDXA2cCl5f94s/XVZoz/ucCFVfW4IXdSSzq3Y5FT/xtV9c2qOq6q9qY3Y+xw4KZmT0xtpKp6FvCn9IYtXpPk00ncgk6bgn8HPs2jk95GfhkFu1wmaMZNHwH8Br0HesdX1RXtVtUNzUJIHwZeX1Vbtl2PBJDkSVV1X9t1DIJ36I0kRyX5HPBJev3nR1TVQYb57CR5apI3JvksvZ11fgDs03JZEkle2EzYuqU5fk6SU1oua1a8Q280D0VvAL7bnHrMG1NVdr1shCS3AZ8Czquqr7dcjrROkiuB1wErqmpJc+6Gqnp2u5VtPB+KPuqAtgvoqF/ywbI2VVV1R/KYAW0Pt1XLIBjojar6EqzbQf2Xm9OrJu6wrg22OMmxwCL6/r65los2AXc0S2ZXkq2BdwA3t1zTrNjl0kiyFfDnwFH0ul0C7ERvneT3VNVDU3y51iPJN4GP0RsKuu7up6quaa0oiXUP6f+G3qJnAT4P/N4oL3RmoDeS/BW9IUt/UFU/ac49FfgQvV1N3tFmfaNqfC2XtuuQJkqyX1V9dbpzo8RAbyT5DvCsif29zRThW6pqcTuVjbYkf0Zvnen/Q28GLuByr2pfkm9U1XOnOzdK7EN/VE328K6qHk7i/3ob743Nx3f3nSt6y8BKcy7JC4EXAQsmrOH0VGCk50cY6I+6KckbJi5u38xqvKWlmkZeVe3Sdg3SBFsDT6GXf/0zQ39MbxjjyLLLpZFkR+BC4H56D/AAlgJPBH69qu5c39fq8ZIcWFVfWN/GIW4YorYkubSqXpbkvKo6ou16Bsk79EYT2C9IciC9dUcALq6qS1ssa5S9FPgC8OpJrhW9/zylNvxCM1xxzyRLePzKqiO7Ibx36JI2K0leBxwNvBhYOeHySG8Ib6BrKNwwRJu6JMdX1Ylt1zFIdrloWEZ6GVJ1X1WdmORpwGJgXt/5y9urana8Q5e0WUry2/Sm+y+kt1vZvsDXR7nLxeVzNVRJPpFku77jpyU5vcWSpHHvAJ4PfLeqDqC3a9m9rVY0Swa6hm2vqrp3/KDZgm5Je+VI6zwwvvhekm2q6hZg15ZrmhX70DVsWyR52vheokmejn/vtGkYa357/BTwL0nu4dH9EEaSfegaqiRvAP6E3k5Q0Nva7/1VdVZ7VUmPleSlwLbA56rqwbbr2VgGuoYuye7A+IOmL1TVTW3WI3WVgS5JHeFDUUnqCANdkjrCQJekjnD4mIYiyU/oraq47lRzHHoLID21lcKkDjPQNSyXAj9Pb5ncc6rqey3XI3Weo1w0NEm2BV4DHElv8aNz6YW7+4lKQ2Cga+iSbEEv1P8W+HOXzpWGwy4XDU2zK8wy4FeBr9Dbyu/L7VYldZd36BqKJLfTW7nuHHpb0a3tvz7K23xJmyoDXUOR5Is8dpRLv5He5kvaVBnoktQR9qFrKJK8ZqrrVXXhXNUibS4MdA3Lq6e4VvTGp0saILtcJKkjvEPXUCR551TXHYsuDZ6BrmH5EL2d1D8L/IzeGi6ShshA17AsoTep6FDgGuBs4NKyj08aGvvQNXR9M0ZfDvxRVa1ouSSpk1wPXUOVZAG9u/U9gTHgrnYrkrrLLhcNRZKjgCPorbJ4PnBEVRnm0hDZ5aKhSPIIcAPw3ebUY/6iVdVhc16U1HHeoWtYDmi7AGlz4x26hirJPOCXm8NVVfVAm/VIXeZDUQ1Fkq2SfJDeg9BPAGcCdyT5YJKt261O6iYDXcPyl8DTgV2q6nlV9VzgmcB29CYdSRowu1w0FEm+Azxr4kSiJFsCt1TV4nYqk7rLO3QNS002K7SqHmb9G19ImgUDXcNyU5I3TDyZ5LeAW1qoR+o8u1w0FEl2pLfm+f301nIBWAo8kd5m0Xe2VZvUVQa6hirJgcAezeFNVXVpm/VIXWagS1JH2IcuSR1hoEtSRxjo6rwkX9vA9vsn+fSw6pGGxUBX51XVi9quQZoLBro6L8l/NB/3T/LFJOcnuSXJPyVJc+3g5tw3gNf0fe2Tk5ye5Kok1yY5vDn/N0ne23z+yiSXJ/Hfk1rl8rna3CyhN4zy+8BXgf2SrAT+HjgQWAWc29f+PcAXquqoJNsBVyX5V+CPgauTfBn4W+BVVfXI3P0Y0uN5R6HNzVVVNdaE73XAIuBXgNuq6jvNcgX/2Nf+FcBxSa4DvkhvB6adq+o+4BjgX4CPVNW/z9lPIK2Hd+ja3Pys7/OHmf7fQIDXVtW3Jrm2J7AG+MUB1SbNinfoUm9tmUVJntkcL+u7dgnw9r6+9iXNx2cA76LXhXNIkhfMYb3SpAx0bfaaXZSWA59pHor2b2Z9IrA1cH2SG4ETm3A/DTi2qr4PHA38Q7M7k9Qap/5LUkd4hy5JHWGgS1JHGOiS1BEGuiR1hIEuSR1hoEtSRxjoktQRBrokdcR/Av+J1cRT7UR8AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "statistics_2_df.plot.bar()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3aad87c",
   "metadata": {},
   "source": [
    "Estimate gnomAD frequencies for each CLINSIG group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "a6afcce8",
   "metadata": {},
   "outputs": [],
   "source": [
    "benign = 'OMIM_vep_gnomad/benign_vep.vcf'\n",
    "pathogenic = 'OMIM_vep_gnomad/pathogenic_vep.vcf'\n",
    "likely_benign = 'OMIM_vep_gnomad/likely_benign_vep.vcf'\n",
    "likely_pathogenic = 'OMIM_vep_gnomad/likely_pathogenic_vep.vcf'\n",
    "benign = 'OMIM_vep_gnomad/benign_vep.vcf'\n",
    "pathogenic = 'OMIM_vep_gnomad/pathogenic_vep.vcf'\n",
    "likely_benign = 'OMIM_vep_gnomad/likely_benign_vep.vcf'\n",
    "likely_pathogenic = 'OMIM_vep_gnomad/likely_pathogenic_vep.vcf'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "c62ffced",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_header(file_name): # header creation function\n",
    "    header_list = []\n",
    "    with open(file_name, \"r\") as file:\n",
    "        while True:\n",
    "            line = file.readline()\n",
    "            if not line:\n",
    "                break\n",
    "            if line.split()[0] == \"#Uploaded_variation\": # save the title\n",
    "                header_list.append(line.split())\n",
    "                return header_list[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "bad4a5da",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_gnomAD_AF_df(name): # dataframe preparation function\n",
    "    df_benign = pd.read_table(name, header=None, skiprows=[i for i in range(0,54)])\n",
    "    df_benign.columns = make_header(name)\n",
    "    df_2 = df_benign.iloc[:, [0,1,2,3,13]]\n",
    "    gnomAD_AF = df_2[df_2['Extra'].str.contains(\"gnomAD_AF\")==True]\n",
    "    return gnomAD_AF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "9eca9e85",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_freq_df(name): # freq_df creation function\n",
    "    df = make_gnomAD_AF_df(name)\n",
    "    freq_list = []\n",
    "    for row in df.itertuples(): \n",
    "        freq_list.append(float((row[5].split(';')[5].split('=')[1])))\n",
    "    freq_df = pd.DataFrame(freq_list, columns=[name.split('/')[1].split('.')[0]])\n",
    "    return freq_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "e3d6e960",
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_freq_df = make_freq_df(benign)\n",
    "likely_benign_freq_df = make_freq_df(likely_benign)\n",
    "pathogenic_freq_df = make_freq_df(pathogenic)\n",
    "likely_pathogenic_freq_df = make_freq_df(likely_pathogenic)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "22fa3c57",
   "metadata": {},
   "source": [
    "Remove zero"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "7574f2ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "benign_freq_non_zero_df = benign_freq_df.loc[benign_freq_df['benign_vep'] != 0]\n",
    "likely_benign_non_zero_df = likely_benign_freq_df.loc[likely_benign_freq_df['likely_benign_vep'] != 0]\n",
    "pathogenic_non_zero_df = pathogenic_freq_df.loc[pathogenic_freq_df['pathogenic_vep'] != 0]\n",
    "likely_pathogenic_non_zero_df = likely_pathogenic_freq_df.loc[likely_pathogenic_freq_df['likely_pathogenic_vep'] != 0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5401dda7",
   "metadata": {},
   "source": [
    "Descriptive statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "5f90829c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>benign_vep</th>\n",
       "      <th>likely_benign_vep</th>\n",
       "      <th>pathogenic_vep</th>\n",
       "      <th>likely_pathogenic_vep</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>116.000000</td>\n",
       "      <td>275.000000</td>\n",
       "      <td>94.000000</td>\n",
       "      <td>100.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>0.035397</td>\n",
       "      <td>0.052555</td>\n",
       "      <td>0.004591</td>\n",
       "      <td>0.000109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>0.133527</td>\n",
       "      <td>0.127765</td>\n",
       "      <td>0.007252</td>\n",
       "      <td>0.000196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>0.000055</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>0.000029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>0.000089</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0.000087</td>\n",
       "      <td>0.000064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>0.001440</td>\n",
       "      <td>0.039880</td>\n",
       "      <td>0.013380</td>\n",
       "      <td>0.000069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>0.618000</td>\n",
       "      <td>0.708700</td>\n",
       "      <td>0.023290</td>\n",
       "      <td>0.001731</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       benign_vep  likely_benign_vep  pathogenic_vep  likely_pathogenic_vep\n",
       "count  116.000000         275.000000       94.000000             100.000000\n",
       "mean     0.035397           0.052555        0.004591               0.000109\n",
       "std      0.133527           0.127765        0.007252               0.000196\n",
       "min      0.000004           0.000008        0.000029               0.000029\n",
       "25%      0.000055           0.000030        0.000042               0.000029\n",
       "50%      0.000089           0.000192        0.000087               0.000064\n",
       "75%      0.001440           0.039880        0.013380               0.000069\n",
       "max      0.618000           0.708700        0.023290               0.001731"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnomAD_freq_df = pd.concat([benign_freq_non_zero_df, likely_benign_non_zero_df, pathogenic_non_zero_df, likely_pathogenic_non_zero_df], axis=1)\n",
    "gnomAD_freq_df.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35ed93e7",
   "metadata": {},
   "source": [
    "Boxplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "287576bb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAWsAAAFUCAYAAADiah0AAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/YYfK9AAAACXBIWXMAAAsTAAALEwEAmpwYAAAvI0lEQVR4nO3de5hcVZnv8e8vIUAgCkLAS5pMwBCUgwwDPeE4I5lEDRLkpoMiOtoogoyG6HDUUYdLENRxZlQMoBiBQwsMiBwYCdCQIERQcSBcDJcAaTBAw3AJ90CAhLznj7UbKpWq7q7quu7+fZ6nnq69a1/eWl399qq1115LEYGZmbW2Uc0OwMzMBudkbWbWBpyszczagJO1mVkbcLI2M2sDTtZmZm3AydpalqRFkkLSl8u8Pjd7vf/xgqTlkv5T0oeqPOe7JF0r6fnsmAcN602Y1chGzQ7ArBRJHcD7s8XPAD8eYPP3Aa8BmwHbAwcDV0k6D+iKiHUVnPqHwA7Ax4FngXsri9ysPpysrVV9mvTN70pgX0m7RMSdZbb974hYW7B8lqR/IiXe24EfVHDedwPXR8RVA20kaZOIeKWC45oNi5tBck7SoZLukfSypDskHSBpsaTF2evTs6/7B0g6TdLK7HGepC2LjvXmbJtHJb0i6V5J/yRJBdv0H+8gST+T9LSkZyWdImm0pL+W9DtJL0q6a4Dmii7gLuArBctDFhE/Am4DSjahlCin6ZICmAR8ur9pJXutv7llF0lXS1oFXJS9tpmk70v6s6RXs5//ImlU0fH/StIN2e/hEUnHSTqx/xzZNpOy8xxWKjZJ04vWf1TSHyW9lJXxryRNLNpmRfa7/ISkZVm5L5H0vhJl8HdZ09Nz2XZ/knR49toCSbeV2Gd7SeskHTWUcrbqOVnnmKSZwPnAPcBHgf8ATgGmlNj8x0AAnwROBP6egqaHLPlcAXyWVFPdH7iKVHv9TonjnQK8CBwCnEpKmqcAvwDOzuJ5GrhE0viiuPcEdgLOjYjlwI3ApySNrqgAoAfYrjiBlXEr8F7gSVJt/r3Zo9Cvgd8CBwA/krQRcDXweVJZzQLOBI4D/r3g/YwHrgXGk/7pfAnYB/hche/ndVly/H/A3aRmny8AuwC/lfSmos33Av5PFtchwGjg8sJ/xpIOBH4DbJwd60DS7+kvsk1+CuwmaWrRsY8k/Z7Pr/a92BBFhB85fQB/AO4EVLBuD1JSXpwtT8+Wu4v2PQ14uX9fYL9su8OKtjsTeAUYX3S8s4u2uzVb/76Cdbtm67qKtv0JqQ16Qrb8hWy7fYq2m5ut36jM++/fb88KyqwPOKfMeb5ctP7T2fppRev/BXgV2DZb/k62vF3BNpsDK9Of4OvrJpUp4/4ynZ4tjwOeK1HG22fn+UrBuhXAM8BbCtZ1Zsf7ZLasbLslwKgy5TIKuB84q2DdGOAx4Ixmf9ZHwsM165zKaqGdwP+L7C8LICJuAf5cYpcripbvADYB3potTwPWAf9ZtN15pNpYcS20p2j5HuDFiPhd0TqA7Qri3gT4BHBtRDySrf4l6R9CRU0hpCQEKTHVwqVFy/sADwJ/kLRR/wNYSEpk/zvb7r3AHyPi4f4dI+JFYEGVcbwXeDNwftF5HyaV6bSi7W+MiGcKlu/IfvZ/49iJVIM+M8pcjM3W/wz4hKQtstUHkT4fP6vyfVgFnKzzazwpYTxR4rXHS6x7umi5/+LZptnPrYCnI+LVou0eK3i90DNFy6+Sele8ruBYmxas3h94C3CppC0LvqpfDRwo6c0lYi+n/5/A/1Swz0CKj7MtKcmtKXrclL2+dfbz7ZQu81LrhmLb7Oc1Jc79noLz9lvvdxtvXBjtL/f+7fsGOe9ZpCaUT2fLRwE3RcQGbdlWe+4Nkl8rSX+825Z47a3AQxUe72lgK0kbFyXstxW8Xgv9tefTs0exj5OaXoZiX+ChwhrtMBXX0J8ifUv5eJntV2Q//4c3vqEUKl73cvZz46L1xcn3qeznYaSLsMVeKBNPOSuznxMG2iginpJ0EfAFSVcDM0jt9dYArlnnVES8RmqD/Pui3hp7kNo2K/Vb0uflY0XrP0WqNd9YZaivk7QtqWnh16REUPx4jCE2hSh13duNdAG0Xq4i1d5XRcSSEo/+JHgj8L8lFTb3bE76FlHocdI3ml2K1n+4aPkPpIQ8ucx5K+0bfh/pH8vnCz8rZfwki+9MUrv5hRWey6rkmnW+nUBqP71U0nxS08hcUtKr5EYRSG3QvwPOkLQNqUa3L6lm9b2CxDQcnyJ9Jn8UEb8tflFSN/B1STtExAMFL+0p6TXS1/odSL0jZgHdwLwaxFXO+aTeMb+R9APgT6Ra8TtJPUYOioiXgB8BXwQWSppLSshfA1YXHiwiQtIvgcMl3Ue6IefDpAuMhds9L+lrwOnZ76KHlDgnAH9HunhcfG2hrOy8XwEuAa6VdAapV8y7SRdJTyjY9o9ZF75pwKnZ+7MGcM06xyJiESkBvpt0ceyfSV24HiP9cVdyrHWkxNGdHeeKbPkYUu+HWugi9Ti4vszrZ5MuGn6maP3vSLXXK4BvkdrG94mIwwovrtZaRKwBPgT8nNSF7UpSAu8i1X5fzbZbCXyA1NzQTWreuSp7P8W+TEqac0kXVjcFji5x7p+R/iHsBJybnXsu6Z/d7VW8l18DM7PFs4DLsve0osTmv8p++sJiA6mOn2VrQUq3cfcC34mIk5odz0iW1bJPiIjBmh5aiqTfA+siYq9mxzKSuBkkxySNJbXZXkOq1e0AfB14iaFfpDPr71K5O/BB4G9IN81YAzlZ59trpN4ap5F6FLwI3AB8LCJq1Z2tLWT9zgeqwa4r18fYgNT98A+kJqbvRsRlzQ1n5HEziI0Iklbwxq3TpZwYEXMbE41Z5VyztpFif9IdmeU82qhAzKrhmrWZWRsYETXr8ePHx6RJk5odhpnlzC233LIyIrZpxLnaLllnd379hNSHdXFEDDo046RJk1iyZEndYzOzkUXSg406V0vcFCPpbElPSLqzaP0+SgPc90r6Rrb6o8DFEXEE6aYAM7Pca4lkDZxDGhPidVlXq9NJtw3vDBwqaWeggzQUJKSuaWZmudcSyToirmfDUdumAr0R8UA2ytuFpI74faSEDS0Sv5lZvbVyspvAGzVoSEl6AmnchL+X9FMGGLxd0pHZXHNLnnzyyfpGamZWZ213gTGbYeOzQ9huPjAfoLOz0/0TzayttXLN+hEKpnsiNX08UmbbkiTtL2n+c89VNMCcmVnLaeVkfTOwYzbV/cakefkqGo8gIhZExJFbbLHF4Btb01xzzTVMmzaN6667rtmhmLWslkjWki4gjUe8k6Q+SYdHxFpgNmnuvWXARRFRagoja3Pf/e53ATjpJI/YalZOS7RZR8ShZdZfSRpUvSqS9gf2nzx5crWHsDq75pprWLt2LQBr167luuuuY8aMGU2Oyqz1tETNul5arRnEX/c31F+r7ufatVlpuU7WrcZf9zfUX6sut2xmSa6TdSv1Bin1dd9go402GnDZzJJcJ+tWagbx1/3SvvWtb623fNxxxzUpErPWlutk3Ur8db+03Xbbbb3lXXfdtTmBmLW4XCfrVmoG8df90rq7uxk9ejQAo0ePpru7u8kRmbWmXCfrVmoG8df90hYtWsRrr6XBE1977TUWLlzY5IjMWlOuk3Ur+eAHP7jesvsSJzNnzlxvee+9925SJGatLdfJupWaQW666ab1lm+55ZYmRdJaxo0bt95yK3wLMmtFI2LC3M7Ozmj2tF777rsvq1aten153LhxXHll1Tdn5sa0adM2WHf99dc3IRKzykm6JSI6G3GuXNesW0lhoi61bGY2ECfrBin+ul+8bGY2ECfrBpk7d+56y74pxswqketk3UoXGKdOnfp6bXrcuHHsscceTY6oNUyfPn29ZfeSMSst18m6lfpZQ6pdjxo1yrXqAnPmzBlw2cySXCfrVjN16lQWL17sWnWB8ePHv167njFjBltvvXVzAzJrUb7n2Zpuzpw5PPPMM65Vmw3Aydqabvz48Zx66qnNDsOspbkZxMysDeQ6WbdSbxArb/bs2UybNo2vfOUrzQ7FrGXlOlm3Wm8QK23p0qUA3HrrrU2OxKx15TpZW+ubPXv2esuuXZuV5mRtTdVfq+7n2rVZaU7WZmZtwMnazKwNOFlbUxVPkLv77rs3KRKz1uZkbU112mmnrbd8yimnNCcQsxaX62Ttftbtob927Vq1WXme1svMrEqe1svMzNbjZG1m1gacrM3M2oCTtZlZG3CyNjNrA07WZmZtwMnazKwNOFmbmbWBtkvWknaQdJaki5sdi5lZozQ0WUs6W9ITku4sWr+PpHsl9Ur6xkDHiIgHIuLw+kZqZtZaGj27+TnAacAv+ldIGg2cDswE+oCbJV0GjAa+V7T/5yLiicaEambWOhqarCPiekmTilZPBXoj4gEASRcCB0bE94D9GhmfmVmranTNupQJwMMFy33AnuU2lrQ18B3gryR9M0vqpbY7EjgSYOLEibWL1sqaN28evb29Fe/X19cHQEdHR0X7TZ48mTlz5lR8PrN21ArJuiIR8RRw1BC2mw/MhzTqXr3jsuqtXr262SGYtbxWSNaPANsVLHdk64ZN0v7A/pMnT67F4WwQ1dZy+/ebN29eLcMxy5VW6Lp3M7CjpO0lbQx8ArisFgeOiAURceQWW2xRi8OZmTVNo7vuXQDcCOwkqU/S4RGxFpgNXA0sAy6KiLsaGZeZWatrdG+QQ8usvxK4stbnczOImeVFKzSD1I2bQcwsL3KdrM3M8iLXydqzm5tZXuQ6WbsZxMzyItfJ2swsL3KdrN0MYmZ5ketk7WYQM8uLXCdrM7O8yHWydjOImeVFrpO1m0HMLC9ynazNzPLCydrMrA20wnjW1mKqnfGlWsuXLweqHw+7Up5hxtpRrpO1R92rTm9vL/fdeSsTx73WkPNtvCZ9wXt5xc11P9dDq0bX/Rxm9ZDrZB0RC4AFnZ2dRzQ7lnYzcdxrHNu5qtlh1NzJS8Y1OwSzqrjN2sysDThZm5m1ASfrBlq5ciVHH300Tz31VLNDMbM242TdQN3d3SxdupTu7u5mh2JmbSbXybqVbjdfuXIlPT09RAQ9PT2uXZtZRXKdrFvpdvPu7m4iAoB169a5dm1mFcl1sm4lixYtYs2aNQCsWbOGhQsXNjkiM2snTtYNMnPmTMaMGQPAmDFj2HvvvZsckZm1k1zfFNNKurq66OnpAWDUqFF0dXU1OSJrhGpv3e/r6wOgo6Oj4n19O30+uWbdIOPHj2fWrFlIYtasWWy99dbNDsla2OrVq1m9enWzw7AW4pp1A3V1dbFixQrXqkeQamu4/fvNmzevluFYG3OybqDx48dz6qmnNjsMM2tDuW4GaaV+1uA7GM2serlO1q3Uzxp8B6OZVS/XybqV+A5GMxsOJ+sG8R2MZjYcTtYN4jsYzWw4nKwbZObMmYwalYp71KhRvoPRzCriZN0gXV1drFu3DkjNIO5rbWaVcD/rBjnvvPPWW77ggguYPXt2k6KxSuV9xnfwbeqtzsm6QS655JL1li+66CIn6zbS29vLbXfdBls26ITpSxi3PXJbY873bGNOY9VzsjYbqi1h3fR1zY6iLkYtdotoq/NvyMysDQyarCXtKqmlkrqkgyT9XNIvJbVFt4r+sazLLZuZDWQoSfg2YHz/gqQrJL292hNKOlvSE5LuLFq/j6R7JfVK+sZAx4iI/4qII4CjgEOqjaWR+vtYl1s2MxvIUJK1ipanAWOHcc5zgH3WO4E0GjgdmAXsDBwqaWdJ75F0edFj24Jdj832a3njxo0bcNnMbCANb96IiOuBp4tWTwV6I+KBiHgVuBA4MCLuiIj9ih5PKPk+0BMRt5Y6j6QjJS2RtOTJJ5+s75sagrlz5663fNJJJzUnEDNrS0NJ1pE9itfV0gTg4YLlvmxdOUcDHwQOlnRUqQ0iYn5EdEZE5zbbbFO7SKs0derU12vT48aNY4899mhyRGbWTobSdU/AeZJeyZY3BX4u6aXCjSLigFoHV05EzAPabgqNuXPn8vWvf921ajOr2FCSdfHwcOeV3Gp4HgG2K1juyNYNi6T9gf0nT5483EPVxNSpU1m8eHGzwzCzNjRoso6IzzYgjpuBHSVtT0rSnwA+OdyDRsQCYEFnZ+cRwz2WmVkzDfsCo6TpkoZc25Z0AXAjsJOkPkmHR8RaYDZwNbAMuCgi7qpBbC01rZeZWbWqut1c0tuAzwKfA7YHfjfUfSPi0DLrrwSurCaeAc7lmrWZ5cKQa9aSRkk6QNJlpJ4bJwPnAh0RMb1O8ZmZGUO73XyypH8ldaf7EXAr6caVdcDFEfFYfUOsnptBzCwvhlKzvpt0u/nHI+KdETE3IpbXOa6aaLXZzc3MqjWUNutlpNvAn5H0fEQsrXNMZmZWZNCadUT8JXAQ8CbgeklLJX2t/+U6xjZsbgYxs7wY0gXGiLg5Io4C3gH8EDgAGA2cJelLwxmFr57cDGJmeVFRP+uIeCkizomIvYB3k7rsHcf643pYGffddx+zZs1q6Fx+ZpYPVd8UExH3RsTXSbeGf6x/vaRvSNqyBrHlzsknn8yLL77It7/97WaHYmZtZth3MEbE2oi4tGDVt4CthnvcWmilNuv77ruPFStWALBixQrXrs2sIvUYz7p4soKmaaU265NPPnm9ZdeuzawSLTW3Yp7116rLLZuZDcTJukEmTZo04LKZ2UBynaxbqc362GOPXW/5+OOPb1IkZtaOcp2sW6nNesqUKa/XpidNmkSrTIhgZu2hHsn6BmB1HY7b9o499lg233xz16rNrGJDHs9a0ptJk9TuQLrN/AHgmoh4oXC7iNi3phHmyJQpU+jp6Wl2GGbWhoaUrCV9HDgD2LLopWclfSEiflXrwMzM7A1DGc/6PaRJcnuAPYCxwGbAX5Om4Tov28bMzOpkKG3Wc4CeiPhURNwWEa9ExMsRcUs2RddV2TYtp5V6g5iZDcdQkvVewE8HeP2n2TYtp5V6g5iZDcdQkvUE0gQE5dxDGszJzMzqZCjJenMG7oq3mtSObWZmdTLUrnu7Snq6zGvjaxWMmZmVNtRkfTUDj6bX0tN7mZm1u6Ek6+3rHkWbmTdvXlXjUff19QHQ0VF5E//kyZOZM6clO92YWQMMZcLcBwd7AGMaEGvbW716NatX+058s2qtXLmSo48+mqeeeqrZoTTckG83LyZpU+Bg4AjgfaQJdFuKpP2B/Ws9aFK1Ndz+/ebNm1fLcMxGjO7ubpYuXUp3dzfHHHNMs8NpqIoHcpK0u6TTgUeBfwfuIo0Z0nLcz9osP1auXElPTw8RQU9Pz4irXQ8pWUvaQtIXJd0K/BbYAngT8IGI+GJEXFfPIM3Muru7iUh9GdatW0d3d3eTI2qsoYwNci7QB3wU+DHwtoj4h3oHZmZWaNGiRaxZswaANWvWsHDhwiZH1FhDqVkfCswDPhYR3RHxYp1jMjPbwMyZMxkzJvVlGDNmDHvvvXeTI2qsoSTrjwO7AX2SLpX0EUnu/WFmDdXV1YWUbvcYNWoUXV1dTY6osQbtDRIRlwCXSOoAPgv8B3AmKdH/laRl0d+Q1Gaq7S9dreXLlwPV9yaphvtnW16MHz+eWbNmcdlllzFr1iy23nrrZofUUEPuuhcRfcBJwEmSPgh8npS0fyDp0oj4xzrFWDe9vb3cdsfdrNtsq4acT6+m/2m33P9YQ8436qVyIwSYtaeuri5WrFgx4mrVUGU/64i4BrhG0luAzwCfq2lUDbRus614eef9mh1GXWx69+XNDsGspsaPH8+pp57a7DCaYlgT5kbEMxHx44j4y1oFZGZmG6qoZi3pI8AMYFvWT/QREYfUMjAzM3vDkGvWkn4A/BLon2/xtYLHutqHVjaOd0s6Q9LFktqundzMrBqV1Ky7SH2tf13tySSdDewHPBERuxSs34d0w81o4MyI+Ndyx4iIZcBRkkYBv2DgKcfMzHKhkjbrl0hTeA3HOcA+hSskjQZOB2YBOwOHStpZ0nskXV702Dbb5wDgCuDKYcZjZtYWKknW/wp8XVLVI/VFxPVAcX+yqUBvRDwQEa8CFwIHRsQdEbFf0eOJ7DiXRcQs4FPlziXpSElLJC158sknqw3ZzKwlVJJ4fw7sDzwi6T5gTeGLEfH+KmOYADxcsNwH7FluY0nTSeOUbMIANeuImA/MB+js7GzLm3bMzPpVkqzPII1bfRXwOE2ayisiFgOLm3FuM7NmqSRZHwJ8JCIW1TiGR4DtCpY7snXDVq/JB8zMGq2SNusnqVESLXIzsKOk7SVtDHwCuKwWB/bkA2aWF5Uk6xOAb0saV+3JJF0A3AjsJKlP0uERsRaYTZpBfRlwUUTcVe05is63v6T5zz33XC0OZ2bWNJU0g3wNmAQ8LukhNrzAuOtgB4iIQ8usv5I6dMOLiAXAgs7OziNqfWwzs0aqJFlfXLcozMxsQJUMkXpiPQOpB19gNLO8qGZ28/dLmi3pS1mf55blC4xmlhdDrllLmgBcCuwBPJqtfoekJaQufY+W3dnMzIalkpr1PNIIe5MjYruI2A7YMVs3rx7BDZd7g5hZXlSSrGcCX4qIP/eviIgHgDnZay3HzSBmlheVtlmXusXc426YmdVZJcn6N8Cpkl6/NVzSROCU7DUzM6uTSpL1HGBz4AFJD0p6ELg/WzenHsENl9uszfLlhBNOYNq0aZx00knNDqXhhpysI+JhYHdgX+A/ssesiNg9IvrqFN+wuM3aLF+uu+46ABYtqvV4cq2vojbrSBZFxKnZ45p6BWZmVuiEE05Yb3mk1a4rnd18T+ADbDi7ORHRkk0hZpYP/bXqfosWLeK4445rUjSNV8lNMV8F/g3oJd0UU9gLpCV7hPh2czPLi0pq1l8G5kTEafUKptY86p6Z5UUlbdZvxrOJm1mTzJgxY73lmTNb8l68uqkkWV8A7FOvQMzMBnLiiesP/DmS2quhsmaQh4ETJf0tsJQNJx/4YS0DMzMrNmPGDK677roRV6uGypL154FVwN9kj0IBOFmbWV2deOKJG9SwR4pKJh/Yvp6B1IN7g5hZXlTSde/sMi8F8DKpS98vW2lca/cGMbO8qKQZZBtgL2AdcGe2bhdAwC3AR0mzn+8VEbfXMkgzs5Gukt4gvwd6gI6ImBYR04AOUne+hcBfAFcAP6h5lGZmI1wlyfrLwLcj4qX+Fdnz7wD/FBGvAt8HdqtphGZmVlGyHge8vcT6t2WvATxPheONmJnZ4CpJ1pcCZ0n6mKRJ2eNjwFnAJdk2U4H7ah2kmRnAypUrOfroo3nqqaeaHUrDVZKsjwKuBs4jTTpwf/b8KuCL2TbLAPe8MLO66O7uZunSpXR3dzc7lIarpJ/1S8BRkv4P8M5s9f0R8WLBNrfXNrzhcT/r6vT19fHiC6M5ecm4wTduMw++MJrN+1pyrgwbxMqVK+np6SEi6Onpoauri6233rrZYTVMpRPmEhEvRsTS7PHi4Hs0j2eKMcuP7u5uItJozOvWrRtxtWtfDLQNdHR08PLa/+HYzlXNDqXmTl4yjk07OpodhlVh0aJFrFmThiRas2YNCxcu5JhjjmlyVI1Tcc3azKwZZs6cyejRowEYPXo0e++9d5MjaiwnazNrC11dXaxbtw5IzSBdXV1NjqixnKzNrC08/fTTr7dZRwTPPPNMkyNqLCdrM2sLJ5988nrL3/72t5sUSXM4WZtZW1ixYsWAy3nnZG1mbWHSpEkDLuedk7WZtYVjjz12veXjjz++SZE0h5O1mbWFKVOmvF6bnjRpEiPtzuS2TNaSNpe0RNJ+zY7FzBrn2GOPZfPNNx9xtWpocLKWdLakJyTdWbR+H0n3SuqV9I0hHOqfgYvqE6WZtaopU6bQ09Mz4mrV0Pjbzc8BTgN+0b9C0mjgdGAm0AfcLOkyYDTwvaL9Pwf8JXA3sGkD4jUzawkNTdYRcb2kSUWrpwK9EfEAgKQLgQMj4nvABs0ckqYDmwM7A6slXRkR6+oZt5nV1rx58+jt7a14v75sxMSOKsZ3mTx5MnPmzKl4v1bRCgM5TQAeLljuA/Yst3FE/AuApMOAleUStaQjgSMBJk6cWKtYzayJVq9e3ewQmqYVknVVIuKcQV6fD8wH6OzsjFLb9PX1Meql59j07strH2ALGPXSU/T1rW12GLnQ19cHz8GoxW15TX5wz0JfNG6c72pruP37zZs3r5bhtIVW+OQ9AmxXsNyRrRs2SftLmv/cc8/V4nBmZk3TCjXrm4EdJW1PStKfAD5ZiwNHxAJgQWdnZ8mpxjo6Onj8lY14eed89gDc9O7L6eh4W7PDyIWOjg6e1JOsm57PyyOjFo+iY4LH+W5lje66dwFwI7CTpD5Jh0fEWmA2aX7HZcBFEXFXjc7nmrWZ5UKje4McWmb9lcCVdTjfgDVrM7N20QrNIGbWpqrtglet5cuXA9VfoKxGq3T5y3Wy9uzmZvXV29vLPbffTqOujPS32z57++0NOd9jDTnL0LRCb5C68ezmZvXV19dHyX6xdbJ19miU4I0bcZot18nazCwv3AxiZlXr6Ojg2ZUrORw1O5S6OItgyypuba+HXCfrofQGGfXS0w27g1EvP5/i2vTNDTnfqJeehoa1JppZPeU6WQ+m0TXu5ctfAGDHdzYqgb5tRA4laZZHIzpZN7o7zkge18DMhifXFxh9B6OZ5UWuk7W77plZXozoZhCzijzbwCFSV2U/xzXmdDxLGlneWpaTtdkQNP5idLqtescJOzbmhBMa/x6tMrlO1u5nbbXii9HWbG6zNjNrA7lO1mZmeeFkbWbWBpyszczaQK4vMFr1Hlo1mpOXNKbf2OMvpTrDWzer//yGD60azZS6n8Ws9nKdrN0bpDqNLq9Xs25qm06qfze1KTT2/VU7k8pwZkRp9Mwmj5FGp6vEU8CrdYmmvI2pfCzsx4Atax9KVXKdrD0HY3XcTa35xo4d2+wQhqTaf3yr+vpYt3p1jaMZ2CZjx1Y83OmWtE7/81wna7Nma4W5++op7++vlfgCo5lZG3CyNjNrA07WZmZtwMnazKwNOFmbmbWBXCdrzxRjZnmR62TtUffMLC9ynazNzPLCydrMrA04WZuZtQEnazOzNuBkbWbWBpyszczagJO1mVkbcLI2M2sDTtZmZm2g7ZK1pOmSbpB0hqTpzY7HzKwRGpqsJZ0t6QlJdxat30fSvZJ6JX1jkMMEsArYFOirV6xmZq2k0dN6nQOcBvyif4Wk0cDpwExS8r1Z0mXAaOB7Rft/DrghIn4r6a3AD4FPNSBuM7OmamiyjojrJU0qWj0V6I2IBwAkXQgcGBHfA/Yb4HDPAJvUJVAzsxbTChPmTgAeLljuA/Yst7GkjwIfIk08fNoA2x0JHAkwceLEWsRpZtY0rZCsKxIRlwCXDGG7+cB8gM7Ozqh3XGZm9dQKvUEeAbYrWO7I1g2bJx8ws7xohZr1zcCOkrYnJelPAJ+sxYEjYgGwoLOz84haHM8GNm/ePHp7eyveb/ny5QDMmTOnov0mT55c8T5m7arRXfcuAG4EdpLUJ+nwiFgLzAauBpYBF0XEXTU6n2vWbWDs2LGMHTu22WGYtbRG9wY5tMz6K4Er63A+16wbyLVcs/pphTZrMzMbRK6TtZtBzCwvcp2sPbu5meVFrpO1mVle5DpZuxnEzPIi18nazSBmlhe5TtZmZnnhZG1m1gZynazdZm1meZHrZO02azPLi1wnazOzvGiFUffaTqNHlwOPMGc20uW6Zt1qbdYeXc7MqqWI/E+i0tnZGUuWLGl2GGaWM5JuiYjORpwr1zVrM7O8cLI2M2sDTtZmZm3AydrMrA3kOlm3Wm8QM7Nq5TpZ+w5GM8uLXCdrM7O8cLI2M2sDTtZmZm3AydrMrA04WZuZtYERMTaIpCeBB5sdR2Y8sLLZQbQgl8uGXCaltVK5/EVEbNOIE42IZN1KJC1p1MAv7cTlsiGXSWkjtVzcDGJm1gacrM3M2oCTdePNb3YALcrlsiGXSWkjslzcZm1m1gZcszYzawNO1mZmbWDEJ2tJkyTdWYPjdEqaV4uYakHSquznOyRdnD0/TNJpFRxjsaSKukjVqjyzY7VUmfbLyvEdBcsrJI1vUixHSfpMmdea8hmoRruUaTNt1OwA8iIilgAtNytvRDwKHNzsOKrRqmUKHAbcCTza5DiIiDOGsE07fAYOo43KtBlGfM06s5Gk8yUtk3SxpM0k7SHpt5JukXS1pLfD6zWN70u6SdJ9kvbK1k+XdHn2fBtJiyTdJelMSQ9KGp/VOpdJ+nn22kJJY0sFJOldkm4qWJ4k6Y7s+UCx/VjS7cBYSVPL1XQlfVjSjVlce2fPb5X0K0njirb9nKRTCpaPkPSjSspzCHE3rUyzn3+W9Lik1ZIek7SDpOMlPS9pZfa4U9I3gE7gfEm3F5zr6Kz87pD0ruz4W0n6L0lLJf1R0q4DvZfstX/IyuF2ST+TNDpbv0rSdyT9KTvWW7P1cyV9NXs+WdI12Ta3Aip4n4Wfgb2z33//Z+ACSSdlzx+X9HT2Xr+Qbb+HpHuzMrpX0hmSjiz3GcjOd0+Zz8Dxkm7Ojj9fycHtUqaS3lnmPV8o6cMFy+dIOljSaEn/nr3npf1lqvTZvl7SFQVlOnA+jogR/QAmAQH8bbZ8NvA14A/ANtm6Q4Czs+eLgR9kz/cFrsmeTwcuz56fBnwze75Pdvzx2bnWArtlr10E/MMAsd0ObJ89/2fgWGDMILH9PHv+EqmmMgm4M1t3WBbbR4AbgLdkcV0PbF5wnuMLjtcJjAPuB8Zk6/8AvKeC8vzqEOJuZplOzo734Wz9dcBNwFb9ZQqcC3wjK9PFQGfBMVcAR2fPvwicmT0/FTghe/5+4PZB3su7gQUF5fwT4DPZ8wD2z57/G3Bs9nwu8NXs+X8DH8mebwqsKvidFH4Grsze4w3AtsAjwO+A2Vl5/DNwIulbzR+BLwAvk4Zs2ARYBNxLhZ+B7PlWBdudW/Ce2qVMNyvznj8CdGfPNwYeBsYCRxYcd5OsTLcnfbZfBnYARmdlevBAucrNIMnDEfH77Pl5wLeAXYBFkiAV5v8UbH9J9vMW0gez2PtIvzwi4ipJzxS89ueIuH2Q/ftdREpq/5r9PATYaZDYLsh+rgPenD0KvZ+UgPeOiOcl7QfsDPw+O97GwI2FO0TEKknXAvtJWkb64N8xQNzF5TkHuGqQuJtZpjuQ/nC/I+k7pDIbB8wAdgcmkv7Q7spee7bEcQvj/2hBzH+fxXytpK0lvXmA9/IBYA/g5qyMxgJPZK+9ClxecI6ZhSeX9CZgQkRcmh335ewYpWyfvecJwN8Cy0m/m5NJCQngReD57AHpn9e9pH+m1wFzqvgM/AcwQ9LXgc1I/wzvIiXTUlquTAd4vz3AjyVtQvpncX1ErJa0N7Br9u0BYAtgx+zcN0XEA9m5Lsjew8XlTuBknRR3Nn8BuCsi3ltm+1eyn69ReRm+UvD8NdKHp5xfAr+SdAkQEbFc0nsGia34vRQv30/6Q51C+i8vYFFEHDpI3GeS/ondA/zfQbYtFYMGibuZZboDsCYidgOQ9H7gaFIt7C7S+/473khkpW5OGE78/USqnX2zxGtrIqueDfMcAL3A24HPkP5x30D6x7kxMD8irn49IGlx9jR44zMwhlQuA9ngMyBpU1KZdkbEw5Lm8kaZltI2ZZr9c1wMfIhUAbiw4PxHF5YppGYQBv9bXY/brJOJkvqTyCdJX/226V8naYyk/1XB8X4PfDzbd29Sc0PFIuJ+0ofoOFKSgVS7GSi2Q7Kfo4DnSP94Cj1Iqpn8Itvvj8DfSpqcHW9zSVNKxPLfwHak8rmg+PUixeX5uyHEPZh6lukDwMaSjsiWP0WqbQKsyZYPJtWwnyPVrN80hNPdkO3b/8e5MiKeH+C9/AY4WNK22WtbSfqLIb6vF4A+SQdl+24ywOYPkpo65pK+PfyUVMO+FfhHSVtImpJ9DvpzxFRSjXS7bJ/uQUIq9RnoT8wrla6LFF70fIE2KFNlbe9l/BL4LLAX6ZskwNWkMh2THWOKpM2z16ZK2j5rqz6EVEZlOVkn9wJfyr7iv4XULnYw8H1JfyK1c/5NBcc7kXQR507gY8BjbJg0h+qXwD+Qvr4TEa8OEtvLkm4jfW0/vNQBI+Ie0gf+V6Sv9YcBF0haSmoCeVeZWC4Cfh8Rz5R5vV9xef50CHEPpm5lSkrIK4CTJb2cHf9BUlv1X5O++r8d2I9UpucAZ2j9i2GlzCVdnFtKanbpGui9RMTdpDbjhdk+i7LzDtWngTnZvn8gu8BYxtmkmtzY7HEYcBDpd/JYdu6fFRzjZlK78HakNu7zBoml1GfgWVKZ3klKYjcXbH8O7VGmbxtg24Wkb2DXZJ93SN9G7gZuzWL7GW/U4PvLdBnwZ+DSASMZqEHbj6ovWm4CbJQ9fy/ZRZAGnHcxBRdp6nD8y4EP5K1MKbgA14gybdbnYxjxTueNC72DfgYGKs92/HzUu0yH+nCbdX1MBC7Kvt68ChwxyPYtTdKWpAtMf4qI3zQpjDyVaTu+l40k3UdzPwMDaccyrYgHcmoBkk4ntRkW+nFEDHYhr2kkbU1qDyz2gYh4qtHxFGvHMq217GL0uUWrX4mIPWt0/Jb+DNRDvct0wHM7WZuZtT5fYDQzawNO1mZmbcDJ2ppCaVS1r5ZbNrP1OVlbXUh6q9KgUvdLekXSI5J6JO1bZpe/Jt3dVsk5/k7Sb5QGWXopO9f52e3H/QMKhYqG+Mz2u0zSk1lsDygNNvSB6t6tWf05WVvNSZpEuhvuQ8A3gV2BDwJXACWHn4yIJyPipQrOsTPpLrGlpDvqdgGOIt1hWPbuPaVRz64l3YXYP9bKAaTxLuo2drakjaTyg3WYDarZncP9yN+DNKrbI8C4Eq9tmf1cQTayWZnlII1Y9ivSoEIPUDCaHvAVoG+QOCZlx+nMlrcjjTfxwzLbq4L3+DngIdLohgtII8NFwetzSXfqHUYaj+U10uBQE0l3qr2QPS4BOor3KzrXYWQj6BUd+/NZDKuB/wLGN/t370f9Hq5ZW01J2oo06tjpEbGq+PVItxwP1fHAr4G/JN0ifrakidlrj5HGGplRwfEOJg1W9G+lXowsEw4mG/PiTOB0YDfgMtLtzsW2J42L8THSe3iV9H7eSvo2MAN4B/BfVdS6J5FumT+Q9K1lR9It5JZTTtZWa5NJ40ksq8Gxzo2I8yKilzTw0lpgWvbar4D/BK5VGjB/gaRjJG0zwPGmAM9HxGP9KyTtpzQIff9jryHENQdYGBHfj4j7IuLnlB7XYWPg0xFxa0TcSRo3YlfgkxGxJNJMOJ8kDcNaaXv5WNK4zLdFGor0C8D+knas8DjWJpysrdZq2S67tP9JRKwFniQNlk9EvBYRnwU6SJMbPESaNOKeCkfzu45UO54BbE4aZ3sw7yLdfl/ov0ts1xcRjxcsvxt4NCJW9K+INJ7xo6QxxSvxSEQ8VHT+ddk5LIecrK3WlpPaiWuRNNYULQdFn9mIeCQizo2IL5ES3jpS0i7lPuDNyqYTy/Z/Mau5/7kG8RZ7sYJt+5tg1rHhP7wxtQnH2pmTtdVURDxNGv5ytormcoTXB4Wq17mfIQ2iv8F5MxeT/gGUGoi+EveQuhoWmjqE/ZYB78h6ywCgNPHBO0jDaEL69vDWojbs3Uoca4Kk7YrOP4raND9ZC/Koe1YPXyINBr9E0nGk5gyRmhq+SeoRMSxZF7zdSG3F95MGtv8M8B7g+6X2iTQ7yVeA07ILoWeTepm8hXQREFKvjcHMA34n6WukXhjTyKaUGsQ1pLI4X9KXs3Wnkro5XpstLyZNd/UtSReShtIsNTP5aqBb0jGk9uszgCsiYnmJbS0HXLO2msvaYXcnDfT+fVKCupbUn/nIGp3mJtI8fj8ldWO7nnQB7zMRcf4Asf2EdDFvS9LkA8tJ/bX/F3BARNww2Ikj4kbSEJxzSO/tINL7HGiOvv7eJgeSas/XZY/HgIP6e6JExDLgH0nltJQ0L+B3SxxuBWnqqAWksn2ANEuJ5ZRH3TOrAUk/Aj4YEe9pwLnmkmbC3qXe57LW4WYQsypkTSCLgFWkfs5HkSaTNasLN4OYFcnGMFlV5tGfkDtJF1LvBL5Maos/pUkh2wjgZhCzIpImkC7alfJ01uPFrKGcrM3M2oCbQczM2oCTtZlZG3CyNjNrA07WZmZtwMnazKwN/H8fy8ecmIg34wAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 360x360 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig, ax= plt.subplots(figsize=(5, 5))\n",
    "ax.set_yscale('log')\n",
    "boxplot = sns.boxplot(x=\"variable\", y=\"value\", data=pd.melt(gnomAD_freq_df))\n",
    "boxplot.axes.set_title(\"gnomAD_frequency\", fontsize=16)\n",
    "boxplot.set_xlabel(\"ClinSIG_group\", fontsize=14)\n",
    "boxplot.set_ylabel(\"gnomAD_AF\", fontsize=14)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "efeb17b2",
   "metadata": {},
   "source": [
    "Calculate the percentage for non-zero gnomAD frequency for each ClinSIG group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "00d953eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "b_per = benign_freq_non_zero_df.shape[0]/benign_freq_df.shape[0]*100\n",
    "lb_per = likely_benign_non_zero_df.shape[0]/likely_benign_freq_df.shape[0]*100\n",
    "p_per = pathogenic_non_zero_df.shape[0]/pathogenic_freq_df.shape[0]*100\n",
    "lp_per = likely_pathogenic_non_zero_df.shape[0]/likely_pathogenic_freq_df.shape[0]*100"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "53282718",
   "metadata": {},
   "source": [
    "Counting the number of each ClinSIG group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "3d8c9df9",
   "metadata": {},
   "outputs": [],
   "source": [
    "b_abs = benign_freq_df.shape[0]\n",
    "lb_abs = likely_benign_freq_df.shape[0]\n",
    "p_abs = pathogenic_freq_df.shape[0]\n",
    "lp_abs = likely_pathogenic_freq_df.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "43c961cd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>clinsig</th>\n",
       "      <th>abs_all</th>\n",
       "      <th>non_zero_percent</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>benign</td>\n",
       "      <td>815</td>\n",
       "      <td>14.233129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>likely_benign</td>\n",
       "      <td>1609</td>\n",
       "      <td>17.091361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>pathogenic</td>\n",
       "      <td>876</td>\n",
       "      <td>10.730594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>994</td>\n",
       "      <td>10.060362</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             clinsig  abs_all  non_zero_percent\n",
       "0             benign      815         14.233129\n",
       "1      likely_benign     1609         17.091361\n",
       "2         pathogenic      876         10.730594\n",
       "3  likely_pathogenic      994         10.060362"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame({'clinsig':['benign', 'likely_benign', 'pathogenic', 'likely_pathogenic'],\n",
    "        'abs_all':[b_abs, lb_abs, p_abs, lp_abs],\n",
    "        'non_zero_percent':[b_per, lb_per, p_per, lp_per]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d416040d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
